% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage{bookmark}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={fig4},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{fig4}
\author{}
\date{\vspace{-2.5em}2025-06-12}

\begin{document}
\maketitle

\begin{verbatim}
## Loading required package: digest
\end{verbatim}

\begin{verbatim}
## Loading required package: tibble
\end{verbatim}

\begin{verbatim}
## Loading required package: ggplot2
\end{verbatim}

\begin{verbatim}
## Project name: 01.protein-seq-evo-v1
\end{verbatim}

\begin{verbatim}
## Loading project configuration
\end{verbatim}

\begin{verbatim}
## Autoloading packages
\end{verbatim}

\begin{verbatim}
##  Loading package: dplyr
\end{verbatim}

\begin{verbatim}
## Loading required package: dplyr
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'dplyr'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:stats':
## 
##     filter, lag
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union
\end{verbatim}

\begin{verbatim}
##  Loading package: stringr
\end{verbatim}

\begin{verbatim}
## Loading required package: stringr
\end{verbatim}

\begin{verbatim}
##  Loading package: readr
\end{verbatim}

\begin{verbatim}
## Loading required package: readr
\end{verbatim}

\begin{verbatim}
##  Loading package: ggplot2
\end{verbatim}

\begin{verbatim}
##  Loading package: tidyr
\end{verbatim}

\begin{verbatim}
## Loading required package: tidyr
\end{verbatim}

\begin{verbatim}
##  Loading package: patchwork
\end{verbatim}

\begin{verbatim}
## Loading required package: patchwork
\end{verbatim}

\begin{verbatim}
##  Loading package: gridExtra
\end{verbatim}

\begin{verbatim}
## Loading required package: gridExtra
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'gridExtra'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:dplyr':
## 
##     combine
\end{verbatim}

\begin{verbatim}
##  Loading package: GGally
\end{verbatim}

\begin{verbatim}
## Loading required package: GGally
\end{verbatim}

\begin{verbatim}
## Registered S3 method overwritten by 'GGally':
##   method from   
##   +.gg   ggplot2
\end{verbatim}

\begin{verbatim}
##  Loading package: readxl
\end{verbatim}

\begin{verbatim}
## Loading required package: readxl
\end{verbatim}

\begin{verbatim}
##  Loading package: viridis
\end{verbatim}

\begin{verbatim}
## Loading required package: viridis
\end{verbatim}

\begin{verbatim}
## Loading required package: viridisLite
\end{verbatim}

\begin{verbatim}
##  Loading package: cowplot
\end{verbatim}

\begin{verbatim}
## Loading required package: cowplot
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'cowplot'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:patchwork':
## 
##     align_plots
\end{verbatim}

\begin{verbatim}
##  Loading package: ggsignif
\end{verbatim}

\begin{verbatim}
## Loading required package: ggsignif
\end{verbatim}

\begin{verbatim}
##  Loading package: minpack.lm
\end{verbatim}

\begin{verbatim}
## Loading required package: minpack.lm
\end{verbatim}

\begin{verbatim}
##  Loading package: purrr
\end{verbatim}

\begin{verbatim}
## Loading required package: purrr
\end{verbatim}

\begin{verbatim}
##  Loading package: scales
\end{verbatim}

\begin{verbatim}
## Loading required package: scales
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'scales'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:purrr':
## 
##     discard
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:viridis':
## 
##     viridis_pal
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:readr':
## 
##     col_factor
\end{verbatim}

\begin{verbatim}
##  Loading package: bigsnpr
\end{verbatim}

\begin{verbatim}
## Loading required package: bigsnpr
\end{verbatim}

\begin{verbatim}
## Loading required package: bigstatsr
\end{verbatim}

\begin{verbatim}
##  Loading package: drc
\end{verbatim}

\begin{verbatim}
## Loading required package: drc
\end{verbatim}

\begin{verbatim}
## Loading required package: MASS
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'MASS'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:patchwork':
## 
##     area
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:dplyr':
## 
##     select
\end{verbatim}

\begin{verbatim}
## 
## 'drc' has been loaded.
\end{verbatim}

\begin{verbatim}
## Please cite R and 'drc' if used for a publication,
\end{verbatim}

\begin{verbatim}
## for references type 'citation()' and 'citation('drc')'.
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'drc'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:stats':
## 
##     gaussian, getInitial
\end{verbatim}

\begin{verbatim}
##  Loading package: data.table
\end{verbatim}

\begin{verbatim}
## Loading required package: data.table
\end{verbatim}

\begin{verbatim}
## 
## Attaching package: 'data.table'
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:purrr':
## 
##     transpose
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:dplyr':
## 
##     between, first, last
\end{verbatim}

\begin{verbatim}
## Autoloading helper functions
\end{verbatim}

\begin{verbatim}
##  Running helper script: globals.R
\end{verbatim}

\begin{verbatim}
##  Running helper script: helpers.R
\end{verbatim}

\begin{verbatim}
## Autoloading data
\end{verbatim}

\begin{verbatim}
## Munging data
\end{verbatim}

\begin{verbatim}
##  Running preprocessing script: 01-util.R
\end{verbatim}

\begin{verbatim}
##  Sourcing R script: 01-util.R
\end{verbatim}

\begin{verbatim}
##  Running preprocessing script: 02-munge_kras.R
\end{verbatim}

\begin{verbatim}
##  Sourcing R script: 02-munge_kras.R
\end{verbatim}

\begin{verbatim}
## Rows: 189 Columns: 8
## -- Column specification --------------------------------------------------------
## Delimiter: "\t"
## chr (6): SEQ, ATOM, COLOR, CONFIDENCE INTERVAL, MSA DATA, RESIDUE VARIETY
## dbl (2): POS, SCORE
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 3591 Columns: 4
## -- Column specification --------------------------------------------------------
## Delimiter: "\t"
## chr (3): X1, X2, X4
## dbl (1): X3
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 3780 Columns: 3
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr (1): variant
## dbl (2): score, pos
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 3572 Columns: 15
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr  (1): mutant
## dbl (11): column_coverage, popEVE, pop-adjusted_ESM1v, pop-adjusted_EVH_epis...
## lgl  (3): pop-adjusted_EVE, pop-adjusted_Tranception, EVE
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 3780 Columns: 43
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr (18): wt_aa, mt_aa, ClinVar_ClinicalSignificance, Starred_Coarse_Grained...
## dbl (11): position, Gold_Stars, NumberSubmitters, frequency_gv2, frequency_g...
## lgl (14): BS1, PM2, PM5, PP5, BP6, b_model, p_model, b_acmg_model, lb_acmg_m...
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 2587 Columns: 28
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr  (8): id, wt_aa, mt_aa, bp_interface, binding_RAF, binding_RAL, binding_...
## dbl (20): Pos_real, abundance_ddg, abundance_ddg_std, pik3cg_ddg, pik3cg_ddg...
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 3453 Columns: 6
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr (2): id, wt_codon
## dbl (4): Pos_real, mean_kcal/mol, std_kcal/mol, ESM1v
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
##  Running preprocessing script: 03-munge_src.R
## 
##  Sourcing R script: 03-munge_src.R
## 
## Rows: 5112 Columns: 9
## -- Column specification --------------------------------------------------------
## Delimiter: " "
## chr (1): id
## dbl (8): FL_activity_mean_kcal/mol, FL_activity_std_kcal/mol, FL_folding_mea...
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 536 Columns: 8
## -- Column specification --------------------------------------------------------
## Delimiter: "\t"
## chr (6): SEQ, ATOM, COLOR, CONFIDENCE INTERVAL, MSA DATA, RESIDUE VARIETY
## dbl (2): POS, SCORE
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 10184 Columns: 4
## -- Column specification --------------------------------------------------------
## Delimiter: "\t"
## chr (3): X1, X2, X4
## dbl (1): X3
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 10720 Columns: 3
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr (1): variant
## dbl (2): score, pos
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 10184 Columns: 15
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr  (1): mutant
## dbl (14): column_coverage, popEVE, pop-adjusted_EVE, pop-adjusted_ESM1v, pop...
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 10184 Columns: 15
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr  (1): mutant
## dbl (14): column_coverage, popEVE, pop-adjusted_EVE, pop-adjusted_ESM1v, pop...
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 536 Columns: 27
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr (15): WT_AA, domains_Pfam, KD_lobe, secondary_structure_uniprot, seconda...
## dbl  (1): position
## lgl (11): block1, block2, block3, block4, block5, R-spine, C-spine, Communit...
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 4898 Columns: 11
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr (2): id, wt_aa
## dbl (9): FL_kinase_fitness_scaled, FL_kinase_sigma_scaled, FL_abundance_fitn...
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
##  Running preprocessing script: 04-munge_psd95.R
## 
##  Sourcing R script: 04-munge_psd95.R
## 
## Rows: 3154 Columns: 41
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr (14): id_eve, id_old, trait_name, library, assay, pdz_name, alignment_po...
## dbl (26): X, V1, pos_am, ddg, std_ddg, ci95_kcal.mol, pdz, structural_alignm...
## lgl  (1): binding_interface_5A
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
##  Running preprocessing script: 05-munge-grb2.R
## 
##  Sourcing R script: 05-munge-grb2.R
## 
## Rows: 1056 Columns: 43
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr (15): id, old_id, id_ref, SS, Pos_class, protein, WT_AA, Mut, wt_aa.x, m...
## dbl (23): Pos_real, Pos_ref, Pos, mut_order, f_dg_pred, f_ddg_pred, f_ddg_pr...
## lgl  (5): f_ddg_pred_conf, b_ddg_pred_conf, allosteric, orthosteric, alloste...
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
## Rows: 1689 Columns: 27
## -- Column specification --------------------------------------------------------
## Delimiter: ","
## chr (12): id, protein, pca_type, aa_seq, old_id, wt_aa.x, mt_aa, wt_aa.y, at...
## dbl (14): Pos_real, Nham_aa, fitness, sigma, growthrate, growthrate_sigma, c...
## lgl  (1): WT
## 
## i Use `spec()` to retrieve the full column specification for this data.
## i Specify the column types or set `show_col_types = FALSE` to quiet this message.
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_abundance }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/data/vampseq/vampseq\_ddg/proteinGym/PTEN\_HUMAN\_Matreyek\_2021.csv"}\NormalTok{, }\AttributeTok{header =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{pten\_activity }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/data/vampseq/vampseq\_ddg/proteinGym/PTEN\_HUMAN\_Mighell\_2018.csv"}\NormalTok{, }\AttributeTok{header =} \ConstantTok{TRUE}\NormalTok{)}

\NormalTok{pten\_esm }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/data/vampseq/vampseq\_ESM1v/P60484.csv"}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_abundance) }\CommentTok{\#5083}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 5083
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nrow}\NormalTok{(pten\_activity) }\CommentTok{\#7260}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 7260
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nrow}\NormalTok{(pten\_esm) }\CommentTok{\#7657}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 7657
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_abundance, pten\_esm, }\AttributeTok{by.x =} \StringTok{"mutant"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"variant"}\NormalTok{) }
\NormalTok{pten\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{ (mutant, DMS\_score, DMS\_score\_bin, ESM}\FloatTok{.1}\NormalTok{v) }\SpecialCharTok{\%\textgreater{}\%} 
\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{rename}\NormalTok{(}\AttributeTok{DMS\_score\_abundance =}\NormalTok{ DMS\_score,}
                \AttributeTok{DMS\_score\_bin\_abundance =}\NormalTok{ DMS\_score\_bin)}
\NormalTok{pten\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_df, pten\_activity, }\AttributeTok{by =} \StringTok{"mutant"}\NormalTok{) }
\NormalTok{pten\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{rename}\NormalTok{(}\AttributeTok{DMS\_score\_activity =}\NormalTok{ DMS\_score,}
                                     \AttributeTok{DMS\_score\_bin\_activity =}\NormalTok{ DMS\_score\_bin) }\SpecialCharTok{\%\textgreater{}\%} 
\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{ (mutant, DMS\_score\_abundance, DMS\_score\_bin\_abundance, ESM}\FloatTok{.1}\NormalTok{v,}
\NormalTok{                 DMS\_score\_activity, DMS\_score\_bin\_activity)}


\NormalTok{pten\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{mutation\_position =} \FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{str\_extract}\NormalTok{(mutant, }\StringTok{"(?\textless{}=}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{D)(}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{d+)(?=}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{D)"}\NormalTok{)))}
\FunctionTok{nrow}\NormalTok{(pten\_df) }\CommentTok{\#4839}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4839
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_clinvar }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/data/vampseq/vampseq\_clinvar/P60484\_clinvar\_PTEN.csv"}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_clinvar) }\CommentTok{\#7657}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 7657
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(pten\_clinvar)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   variant   V1      Model               Dataset  ddG_pred position wildtype
## 1   A120C 2381 ThermoMPNN AF-P60484-F1-model_v2 0.7477777      119        A
## 2   A120D 2382 ThermoMPNN AF-P60484-F1-model_v2 2.9353685      119        A
## 3   A120E 2383 ThermoMPNN AF-P60484-F1-model_v2 2.7981739      119        A
## 4   A120F 2384 ThermoMPNN AF-P60484-F1-model_v2 1.7522278      119        A
## 5   A120G 2385 ThermoMPNN AF-P60484-F1-model_v2 1.6180537      119        A
## 6   A120H 2386 ThermoMPNN AF-P60484-F1-model_v2 2.7476149      119        A
##   mutation                   pdb chain new_position uniprot exposure_SS
## 1        C AF-P60484-F1-model_v2     A          120  P60484           E
## 2        D AF-P60484-F1-model_v2     A          120  P60484           E
## 3        E AF-P60484-F1-model_v2     A          120  P60484           E
## 4        F AF-P60484-F1-model_v2     A          120  P60484           E
## 5        G AF-P60484-F1-model_v2     A          120  P60484           E
## 6        H AF-P60484-F1-model_v2     A          120  P60484           E
##   exposure_rASA spot_disorder func_esms_residue_class func_esms_variant_class
## 1             0             0                       2                       2
## 2             0             0                       2                       2
## 3             0             0                       2                       2
## 4             0             0                       2                       2
## 5             0             0                       2                       2
## 6             0             0                       2                       2
##   clinvar_clinical_significance
## 1                              
## 2                              
## 3                      conflict
## 4                              
## 5                           VUS
## 6                              
##                                                                                                                                                                                                                                                                                                                                                                                                              sequence
## 1 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 2 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 3 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 4 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 5 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 6 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_df, pten\_clinvar, }\AttributeTok{by.x=}\StringTok{"mutant"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"variant"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_df) }\CommentTok{\#4839}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4839
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_var }\OtherTok{\textless{}{-}} \FunctionTok{read\_excel}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/data/vampseq/vampseq\_clinvar/taylor\_supplement.xlsx"}\NormalTok{, }\AttributeTok{sheet =} \StringTok{"Table S3"}\NormalTok{)}
\FunctionTok{head}\NormalTok{(pten\_var)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 6 x 34
##   Subject_ID Nucleotide_Change Amino_Acid_Change AA_one_letter Age_Assessment
##        <dbl> <chr>             <chr>             <chr>                  <dbl>
## 1        330 c.716T>C          p.(Met239Thr)     M239T                     26
## 2        370 c.395G>A          p.(Gly132Asp)     G132D                     63
## 3        435 c.406T>C          p.(Cys136Arg)     C136R                     52
## 4        522 c.103A>G          p.(Met35Val)      M35V                      35
## 5        897 c.284C>T          p.(Pro95Leu)      P95L                      17
## 6       1005 c.144C>A          p.(Asn48Lys)      N48K                      39
## # i 29 more variables: Sex <chr>, CC_Score <chr>, Fitness_Score <dbl>,
## #   Abundance_Score <dbl>, Phenotype_List <chr>, Phenotype_Class <chr>,
## #   ASD <lgl>, PHTS <lgl>, Age_Head_Measure <dbl>, `Head_Measure(cm)` <dbl>,
## #   Head_size_expected <dbl>, Head_size_expected_SD <dbl>, Head_z_score <dbl>,
## #   CADD_Score <dbl>, Fitness_score_coded <dbl>, Abundance_score_coded <dbl>,
## #   Quadrant <dbl>, Thyroid_Age_Dx <chr>, Breast_1_Age_Dx <chr>,
## #   Breast_2_Age_Dx <chr>, Melanoma_Age_Dx <chr>, Colorec_Age_Dx <chr>, ...
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nrow}\NormalTok{(pten\_var) }\CommentTok{\#145}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 145
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_var }\OtherTok{\textless{}{-}}\NormalTok{ pten\_var }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(AA\_one\_letter, Phenotype\_Class, Phenotype\_List, Fitness\_Score, Abundance\_Score)}
\NormalTok{pten\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_df, pten\_var, }\AttributeTok{by.x=}\StringTok{"mutant"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"AA\_one\_letter"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_df) }\CommentTok{\#4873}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4873
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_gnomad }\OtherTok{\textless{}{-}} \FunctionTok{read.csv}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/data/vampseq/vampseq\_gnomad/PTEN\_gnomAD\_v4.1.0\_ENST00000371953\_2025\_04\_08\_14\_29\_46.csv"}\NormalTok{)}
\NormalTok{pten\_gnomad }\OtherTok{\textless{}{-}}\NormalTok{ pten\_gnomad }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(VEP.Annotation }\SpecialCharTok{==} \StringTok{"missense\_variant"}\NormalTok{)}
\FunctionTok{summary}\NormalTok{(pten\_gnomad}\SpecialCharTok{$}\NormalTok{Allele.Frequency)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
## 6.195e-07 6.197e-07 6.211e-07 2.946e-06 1.859e-06 1.289e-04
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#      Min.   1st Qu.    Median      Mean   3rd Qu.      Max. }
\CommentTok{\# 6.195e{-}07 6.197e{-}07 6.211e{-}07 2.946e{-}06 1.859e{-}06 1.289e{-}04 }
\NormalTok{pten\_gnomad }\OtherTok{\textless{}{-}}\NormalTok{  pten\_gnomad }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(HGVS.Consequence, ClinVar.Germline.Classification,}
\NormalTok{                                              rsIDs, ClinVar.Variation.ID)}

\CommentTok{\# Apply to the column}
\NormalTok{aa\_map }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \AttributeTok{Ala=}\StringTok{"A"}\NormalTok{, }\AttributeTok{Arg=}\StringTok{"R"}\NormalTok{, }\AttributeTok{Asn=}\StringTok{"N"}\NormalTok{, }\AttributeTok{Asp=}\StringTok{"D"}\NormalTok{, }\AttributeTok{Cys=}\StringTok{"C"}\NormalTok{,}
  \AttributeTok{Gln=}\StringTok{"Q"}\NormalTok{, }\AttributeTok{Glu=}\StringTok{"E"}\NormalTok{, }\AttributeTok{Gly=}\StringTok{"G"}\NormalTok{, }\AttributeTok{His=}\StringTok{"H"}\NormalTok{, }\AttributeTok{Ile=}\StringTok{"I"}\NormalTok{,}
  \AttributeTok{Leu=}\StringTok{"L"}\NormalTok{, }\AttributeTok{Lys=}\StringTok{"K"}\NormalTok{, }\AttributeTok{Met=}\StringTok{"M"}\NormalTok{, }\AttributeTok{Phe=}\StringTok{"F"}\NormalTok{, }\AttributeTok{Pro=}\StringTok{"P"}\NormalTok{,}
  \AttributeTok{Ser=}\StringTok{"S"}\NormalTok{, }\AttributeTok{Thr=}\StringTok{"T"}\NormalTok{, }\AttributeTok{Trp=}\StringTok{"W"}\NormalTok{, }\AttributeTok{Tyr=}\StringTok{"Y"}\NormalTok{, }\AttributeTok{Val=}\StringTok{"V"}\NormalTok{,}
  \AttributeTok{Ter=}\StringTok{"*"}
\NormalTok{)}
\NormalTok{convert\_to\_short }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(consequence) \{}
\NormalTok{  consequence }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{"\^{}p}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{."}\NormalTok{, }\StringTok{""}\NormalTok{, consequence)  }\CommentTok{\# Remove \textquotesingle{}p.\textquotesingle{}}
\NormalTok{  matches }\OtherTok{\textless{}{-}} \FunctionTok{regmatches}\NormalTok{(consequence, }\FunctionTok{regexec}\NormalTok{(}\StringTok{"([A{-}Za{-}z]\{3\})([0{-}9]+)([A{-}Za{-}z]\{3\})"}\NormalTok{, consequence))[[}\DecValTok{1}\NormalTok{]]}
  \ControlFlowTok{if}\NormalTok{ (}\FunctionTok{length}\NormalTok{(matches) }\SpecialCharTok{==} \DecValTok{4}\NormalTok{) \{}
\NormalTok{    from }\OtherTok{\textless{}{-}}\NormalTok{ aa\_map[[matches[}\DecValTok{2}\NormalTok{]]]}
\NormalTok{    pos }\OtherTok{\textless{}{-}}\NormalTok{ matches[}\DecValTok{3}\NormalTok{]}
\NormalTok{    to }\OtherTok{\textless{}{-}}\NormalTok{ aa\_map[[matches[}\DecValTok{4}\NormalTok{]]]}
    \FunctionTok{return}\NormalTok{(}\FunctionTok{paste0}\NormalTok{(from, pos, to))}
\NormalTok{  \} }\ControlFlowTok{else}\NormalTok{ \{}
    \FunctionTok{return}\NormalTok{(}\ConstantTok{NA}\NormalTok{)}
\NormalTok{  \}}
\NormalTok{\}}
\NormalTok{pten\_gnomad}\SpecialCharTok{$}\NormalTok{ID }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(pten\_gnomad}\SpecialCharTok{$}\NormalTok{HGVS.Consequence, convert\_to\_short)}

\FunctionTok{head}\NormalTok{(pten\_gnomad)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   HGVS.Consequence              ClinVar.Germline.Classification        rsIDs
## 1        p.Ile5Met                                              rs2132145458
## 2        p.Val9Asp                                              rs2132145566
## 3       p.Ser10Gly                       Uncertain significance  rs572685299
## 4       p.Tyr16His                       Uncertain significance rs1064796078
## 5       p.Asp19Gly                       Uncertain significance rs1554890392
## 6       p.Asp22Gly Conflicting classifications of pathogenicity rs1554890398
##   ClinVar.Variation.ID   ID
## 1                   NA  I5M
## 2                   NA  V9D
## 3               428272 S10G
## 4               428195 Y16H
## 5               492333 D19G
## 6               486221 D22G
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nrow}\NormalTok{(pten\_gnomad) }\CommentTok{\#235}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 235
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(pten\_gnomad}\SpecialCharTok{$}\NormalTok{ID))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 235
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_gnomad}\SpecialCharTok{$}\NormalTok{gnomad }\OtherTok{\textless{}{-}} \StringTok{"gnomad"}

\NormalTok{pten\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_df, pten\_gnomad, }\AttributeTok{by.x=}\StringTok{"mutant"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"ID"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_df) }\CommentTok{\#4873}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4873
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(pten\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   mutant DMS_score_abundance DMS_score_bin_abundance    ESM.1v
## 1  A120E        -0.075192482                       0 -16.59294
## 2  A120F         0.449809072                       0 -16.11391
## 3  A120G         0.639477479                       0 -10.35606
## 4  A120L         0.023241040                       0 -14.00481
## 5  A120P        -0.004816027                       0 -14.67169
## 6  A120Q         0.108151133                       0 -19.15990
##   DMS_score_activity DMS_score_bin_activity mutation_position   V1      Model
## 1         -2.0938569                      0               120 2383 ThermoMPNN
## 2         -3.4270689                      0               120 2384 ThermoMPNN
## 3         -0.0142132                      1               120 2385 ThermoMPNN
## 4         -3.4214305                      0               120 2389 ThermoMPNN
## 5         -3.4173247                      0               120 2392 ThermoMPNN
## 6         -4.5183149                      0               120 2393 ThermoMPNN
##                 Dataset ddG_pred position wildtype mutation
## 1 AF-P60484-F1-model_v2 2.798174      119        A        E
## 2 AF-P60484-F1-model_v2 1.752228      119        A        F
## 3 AF-P60484-F1-model_v2 1.618054      119        A        G
## 4 AF-P60484-F1-model_v2 1.529649      119        A        L
## 5 AF-P60484-F1-model_v2 2.719771      119        A        P
## 6 AF-P60484-F1-model_v2 2.936961      119        A        Q
##                     pdb chain new_position uniprot exposure_SS exposure_rASA
## 1 AF-P60484-F1-model_v2     A          120  P60484           E             0
## 2 AF-P60484-F1-model_v2     A          120  P60484           E             0
## 3 AF-P60484-F1-model_v2     A          120  P60484           E             0
## 4 AF-P60484-F1-model_v2     A          120  P60484           E             0
## 5 AF-P60484-F1-model_v2     A          120  P60484           E             0
## 6 AF-P60484-F1-model_v2     A          120  P60484           E             0
##   spot_disorder func_esms_residue_class func_esms_variant_class
## 1             0                       2                       2
## 2             0                       2                       2
## 3             0                       2                       2
## 4             0                       2                       2
## 5             0                       2                       2
## 6             0                       2                       2
##   clinvar_clinical_significance
## 1                      conflict
## 2                              
## 3                           VUS
## 4                              
## 5                              
## 6                              
##                                                                                                                                                                                                                                                                                                                                                                                                              sequence
## 1 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 2 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 3 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 4 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 5 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 6 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
##   Phenotype_Class
## 1            PHTS
## 2            <NA>
## 3            <NA>
## 4            <NA>
## 5            <NA>
## 6            <NA>
##                                                                                                                                                                                                                 Phenotype_List
## 1 Adenomatous Polyps|Anemia|Benign Neurologic Neoplasms|Gastroesophageal Reflux Disease|Hamartomatous Polyp|Hemangioma of Skin|Inflammatory Polyp|Lipoma|Macrocephaly|Malignant Neoplasm of Prostate|Polyp Hyperplastic Benign
## 2                                                                                                                                                                                                                         <NA>
## 3                                                                                                                                                                                                                         <NA>
## 4                                                                                                                                                                                                                         <NA>
## 5                                                                                                                                                                                                                         <NA>
## 6                                                                                                                                                                                                                         <NA>
##   Fitness_Score Abundance_Score HGVS.Consequence
## 1     -2.093857     -0.07519248             <NA>
## 2            NA              NA             <NA>
## 3            NA              NA             <NA>
## 4            NA              NA             <NA>
## 5            NA              NA             <NA>
## 6            NA              NA             <NA>
##   ClinVar.Germline.Classification rsIDs ClinVar.Variation.ID gnomad
## 1                            <NA>  <NA>                   NA   <NA>
## 2                            <NA>  <NA>                   NA   <NA>
## 3                            <NA>  <NA>                   NA   <NA>
## 4                            <NA>  <NA>                   NA   <NA>
## 5                            <NA>  <NA>                   NA   <NA>
## 6                            <NA>  <NA>                   NA   <NA>
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{all\_positions }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \DecValTok{88}\SpecialCharTok{:}\DecValTok{98}\NormalTok{,    }\CommentTok{\# WPD loop}
  \DecValTok{160}\SpecialCharTok{:}\DecValTok{171}\NormalTok{,  }\CommentTok{\# TI loop}
  \DecValTok{123}\SpecialCharTok{:}\DecValTok{130}\NormalTok{,  }\CommentTok{\# P loop}
  \DecValTok{35}\SpecialCharTok{:}\DecValTok{49}\NormalTok{,    }\CommentTok{\# Arginine loop}
  \DecValTok{200}\SpecialCharTok{:}\DecValTok{212}\NormalTok{,  }\CommentTok{\# CBR1}
  \DecValTok{226}\SpecialCharTok{:}\DecValTok{238}\NormalTok{,  }\CommentTok{\# CBR2}
  \DecValTok{258}\SpecialCharTok{:}\DecValTok{268}\NormalTok{,  }\CommentTok{\# CBR3}
  \DecValTok{327}\SpecialCharTok{:}\DecValTok{335}\NormalTok{,   }\CommentTok{\# CÎ±2 loop}
  \DecValTok{151}\SpecialCharTok{:}\DecValTok{174}\NormalTok{, }\CommentTok{\# membrane{-}binding alpha{-}helix}
  \DecValTok{1}\SpecialCharTok{:}\DecValTok{15} \CommentTok{\# PBM motif }
\NormalTok{)}

\FunctionTok{length}\NormalTok{(pten\_df[}\FunctionTok{is.na}\NormalTok{(pten\_df}\SpecialCharTok{$}\NormalTok{DMS\_score\_abundance),] }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{pull}\NormalTok{(mutant)) }\CommentTok{\#}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df[}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(pten\_df}\SpecialCharTok{$}\NormalTok{DMS\_score\_abundance), ]}
\FunctionTok{nrow}\NormalTok{(pten\_df) }\CommentTok{\#4873}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4873
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{fil\_pten\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ all\_positions) }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{(mutant, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}

\FunctionTok{nrow}\NormalTok{(fil\_pten\_df) }\CommentTok{\#3454}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 3454
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Fit a loess model using the filtered data}
\NormalTok{loess\_fit }\OtherTok{\textless{}{-}} \FunctionTok{loess}\NormalTok{(DMS\_score\_activity }\SpecialCharTok{\textasciitilde{}}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{data =}\NormalTok{ fil\_pten\_df, }\AttributeTok{span =} \FloatTok{0.7}\NormalTok{, }\AttributeTok{family =} \StringTok{"symmetric"}\NormalTok{)}

\CommentTok{\# Predict fitted values for ALL data points using the loess model trained on fil\_pten\_df}
\NormalTok{pten\_df}\SpecialCharTok{$}\NormalTok{fitted }\OtherTok{\textless{}{-}} \FunctionTok{predict}\NormalTok{(loess\_fit, }\AttributeTok{newdata =}\NormalTok{ pten\_df)}

\CommentTok{\# Calculate residuals for ALL points}
\NormalTok{pten\_df}\SpecialCharTok{$}\NormalTok{residuals }\OtherTok{\textless{}{-}}\NormalTok{  pten\_df}\SpecialCharTok{$}\NormalTok{DMS\_score\_activity }\SpecialCharTok{{-}}\NormalTok{ pten\_df}\SpecialCharTok{$}\NormalTok{fitted}
\FunctionTok{range}\NormalTok{(pten\_df}\SpecialCharTok{$}\NormalTok{residuals) }\CommentTok{\#{-}5.157748  4.754293}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] -5.157748  4.754293
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{fit\_line\_df }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{DMS\_score\_abundance =} \FunctionTok{seq}\NormalTok{(}\FunctionTok{min}\NormalTok{(fil\_pten\_df}\SpecialCharTok{$}\NormalTok{DMS\_score\_abundance, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{),}
                            \FunctionTok{max}\NormalTok{(fil\_pten\_df}\SpecialCharTok{$}\NormalTok{DMS\_score\_abundance, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{),}
                            \AttributeTok{length.out =} \DecValTok{200}\NormalTok{)}
\NormalTok{)}
\NormalTok{fit\_line\_df}\SpecialCharTok{$}\NormalTok{DMS\_score\_activity }\OtherTok{\textless{}{-}} \FunctionTok{predict}\NormalTok{(loess\_fit, }\AttributeTok{newdata =}\NormalTok{ fit\_line\_df)}

\FunctionTok{range}\NormalTok{(pten\_df}\SpecialCharTok{$}\NormalTok{residuals)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] -5.157748  4.754293
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{range}\NormalTok{(pten\_df}\SpecialCharTok{$}\NormalTok{DMS\_score\_abundance)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] -0.2776983  1.6652985
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{range}\NormalTok{(pten\_df}\SpecialCharTok{$}\NormalTok{DMS\_score\_activity)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] -5.754822  2.817851
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{mutation\_position =} \FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{str\_extract}\NormalTok{(mutant, }\StringTok{"(?\textless{}=}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{D)(}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{d+)(?=}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{D)"}\NormalTok{)))}

\FunctionTok{nrow}\NormalTok{(pten\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4873
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{median\_residuals }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(mutation\_position) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}\AttributeTok{median\_residuals =} \FunctionTok{median}\NormalTok{(residuals, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{))}

\FunctionTok{min}\NormalTok{(median\_residuals}\SpecialCharTok{$}\NormalTok{median\_residuals) }\CommentTok{\#{-}1.861488}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] -4.034455
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{max}\NormalTok{(median\_residuals}\SpecialCharTok{$}\NormalTok{median\_residuals) }\CommentTok{\#4.034455}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1.861488
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pdb }\OtherTok{\textless{}{-}} \FunctionTok{read.pdb}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/data/residual\_pdb/PTEN/1d5r.pdb"}\NormalTok{)}
\NormalTok{data }\OtherTok{\textless{}{-}}\NormalTok{ median\_residuals}
\FunctionTok{head}\NormalTok{(data)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 6 x 2
##   mutation_position median_residuals
##               <dbl>            <dbl>
## 1                 2            0.728
## 2                 3            1.34 
## 3                 5           -0.224
## 4                 6           -1.19 
## 5                 7            0.579
## 6                 9           -0.625
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Create a new B{-}factor vector initialized with the current B{-}factors from the PDB}
\NormalTok{new\_b\_factors }\OtherTok{\textless{}{-}}\NormalTok{ pdb}\SpecialCharTok{$}\NormalTok{atom}\SpecialCharTok{$}\NormalTok{b}

\CommentTok{\# Loop through each position in the correlation data and update the B{-}factors}
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\FunctionTok{nrow}\NormalTok{(data)) \{}
\NormalTok{  position }\OtherTok{\textless{}{-}}\NormalTok{ data}\SpecialCharTok{$}\NormalTok{mutation\_position[i]}
\NormalTok{  correlation\_value }\OtherTok{\textless{}{-}}\NormalTok{ data}\SpecialCharTok{$}\NormalTok{median\_residuals[i]}
  
  \CommentTok{\# Find indices in the PDB that match the current position}
\NormalTok{  indices }\OtherTok{\textless{}{-}} \FunctionTok{which}\NormalTok{(pdb}\SpecialCharTok{$}\NormalTok{atom}\SpecialCharTok{$}\NormalTok{resno }\SpecialCharTok{==}\NormalTok{ position)}
  
  \CommentTok{\# Print the indices and current B{-}factors before updating}
  \CommentTok{\#cat("Updating residue number:", position, "\textbackslash{}n")}
  \CommentTok{\#cat("Indices in PDB:", indices, "\textbackslash{}n")}
  \CommentTok{\#cat("Current B{-}factors:", new\_b\_factors[indices], "\textbackslash{}n")}
  
  \CommentTok{\# Update B{-}factors for all atoms in the current residue}
\NormalTok{  new\_b\_factors[indices] }\OtherTok{\textless{}{-}}\NormalTok{ correlation\_value}
  
  \CommentTok{\# Print the new B{-}factors after updating}
  \CommentTok{\#cat("Updated B{-}factors:", new\_b\_factors[indices], "\textbackslash{}n")}
  \CommentTok{\#cat("\textbackslash{}n")  \# Add an extra line for readability}
\NormalTok{\}}
\CommentTok{\# Replace non{-}matching B{-}factors with outlier value so we can filter it out in ChimeraX}
\NormalTok{non\_matching\_indices }\OtherTok{\textless{}{-}} \FunctionTok{setdiff}\NormalTok{(}\FunctionTok{seq\_along}\NormalTok{(new\_b\_factors), }\FunctionTok{which}\NormalTok{(pdb}\SpecialCharTok{$}\NormalTok{atom}\SpecialCharTok{$}\NormalTok{resno }\SpecialCharTok{\%in\%}\NormalTok{ data}\SpecialCharTok{$}\NormalTok{mutation\_position))}
\NormalTok{new\_b\_factors[non\_matching\_indices] }\OtherTok{\textless{}{-}} \DecValTok{999}

\CommentTok{\# Assign the new B{-}factors back to the pdb structure}
\CommentTok{\# Write the modified PDB structure to a new file}
\NormalTok{pdb}\SpecialCharTok{$}\NormalTok{atom}\SpecialCharTok{$}\NormalTok{b }\OtherTok{\textless{}{-}}\NormalTok{ new\_b\_factors}
\FunctionTok{write.pdb}\NormalTok{(pdb, }\AttributeTok{file =} \StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/data/residual\_pdb/PTEN/1d5r\_median\_residual\_loess.pdb"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_soma }\OtherTok{\textless{}{-}} \FunctionTok{read.delim}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/data/vampseq/vampseq\_somatic/All\_cBioportal\_nonredundant\_210417.tsv"}\NormalTok{, }\AttributeTok{header =} \ConstantTok{TRUE}\NormalTok{, }\AttributeTok{sep =} \StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{)}
\NormalTok{pten\_soma }\OtherTok{\textless{}{-}}\NormalTok{ pten\_soma }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{grepl}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{*"}\NormalTok{, Protein.Change))}
\NormalTok{pten\_soma}\SpecialCharTok{$}\NormalTok{somatic }\OtherTok{\textless{}{-}} \StringTok{"Somatic"}
\FunctionTok{head}\NormalTok{(pten\_soma)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                                                     Study         Sample.ID
## 1                  Acral Melanoma (TGEN, Genome Res 2017)       148336T-25b
## 2       Non-Small Cell Cancer (MSKCC, Cancer Discov 2017) P-0003863-T01-IM5
## 3       Non-Small Cell Cancer (MSKCC, Cancer Discov 2017) P-0009431-T01-IM5
## 4 Uterine Sarcoma/Mesenchymal (MSK, Clin Cancer Res 2020) P-0001821-T01-IM3
## 5                  Pan-Lung Cancer (TCGA, Nat Genet 2016)   TCGA-39-5027-01
## 6                  Pan-Lung Cancer (TCGA, Nat Genet 2016)   TCGA-21-5783-01
##   Cancer.Type Protein.Change
## 1                      R130G
## 2                      R130Q
## 3                      R130G
## 4                      R130Q
## 5                      R130Q
## 6                      R130Q
##                                                                                                         Annotation
## 1                   OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: present;CancerHotspot: yes;3DHotspot: yes
## 2 OncoKB: Likely Oncogenic, level_4;CIViC: Functional: 1;MyCancerGenome: present;CancerHotspot: yes;3DHotspot: yes
## 3                   OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: present;CancerHotspot: yes;3DHotspot: yes
## 4 OncoKB: Likely Oncogenic, level_4;CIViC: Functional: 1;MyCancerGenome: present;CancerHotspot: yes;3DHotspot: yes
## 5 OncoKB: Likely Oncogenic, level_4;CIViC: Functional: 1;MyCancerGenome: present;CancerHotspot: yes;3DHotspot: yes
## 6 OncoKB: Likely Oncogenic, level_4;CIViC: Functional: 1;MyCancerGenome: present;CancerHotspot: yes;3DHotspot: yes
##                                                                                                                         Functional.Impact
## 1    MutationAssessor: impact: high, score: 4.165;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.999
## 2 MutationAssessor: impact: high, score: 3.615;SIFT: impact: deleterious, score: 0.02;Polyphen-2: impact: probably_damaging, score: 0.998
## 3    MutationAssessor: impact: high, score: 4.165;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.999
## 4 MutationAssessor: impact: high, score: 3.615;SIFT: impact: deleterious, score: 0.02;Polyphen-2: impact: probably_damaging, score: 0.998
## 5 MutationAssessor: impact: high, score: 3.615;SIFT: impact: deleterious, score: 0.02;Polyphen-2: impact: probably_damaging, score: 0.998
## 6 MutationAssessor: impact: high, score: 3.615;SIFT: impact: deleterious, score: 0.02;Polyphen-2: impact: probably_damaging, score: 0.998
##       Mutation.Type Variant.Type     Copy.. COSMIC      MS      VS
## 1 Missense_Mutation          SNP    diploid    222    <NA>    <NA>
## 2 Missense_Mutation          SNP    diploid    222 SOMATIC    <NA>
## 3 Missense_Mutation          SNP    diploid    222 SOMATIC    <NA>
## 4 Missense_Mutation          SNP    diploid    222 SOMATIC Unknown
## 5 Missense_Mutation          SNP shallowdel    222 Somatic    <NA>
## 6 Missense_Mutation          SNP shallowdel    222 Somatic    <NA>
##          Center Chromosome Start.Pos  End.Pos Ref Var            HGVSg
## 1             .         10  89692904 89692904   C   G 10:g.89692904C>G
## 2         MSKCC         10  89692905 89692905   G   A 10:g.89692905G>A
## 3         MSKCC         10  89692904 89692904   C   G 10:g.89692904C>G
## 4         MSKCC         10  89692905 89692905   G   A 10:g.89692905G>A
## 5 broad.mit.edu         10  89692905 89692905   G   A 10:g.89692905G>A
## 6 broad.mit.edu         10  89692905 89692905   G   A 10:g.89692905G>A
##                        HGVSc Allele.Freq..T. Allele.Freq..N. Variant.Reads
## 1 ENST00000371953.3:c.388C>G              NA              NA            NA
## 2 ENST00000371953.3:c.389G>A              NA              NA            NA
## 3 ENST00000371953.3:c.388C>G              NA              NA            NA
## 4 ENST00000371953.3:c.389G>A            0.35              NA           174
## 5 ENST00000371953.3:c.389G>A            0.23              NA            17
## 6 ENST00000371953.3:c.389G>A            0.42              NA            39
##   Ref.Reads Variant.Reads..Normal. Ref.Reads..Normal. X..Mut.in.Sample Exon
## 1        NA                     NA                 NA              357  5/9
## 2        NA                     NA                 NA                8  5/9
## 3        NA                     NA                 NA                9  5/9
## 4       318                     NA                275               65  5/9
## 5        56                     NA                 81              295  5/9
## 6        53                     NA                186              363  5/9
##   gnomAD                                                     ClinVar
## 1     NA                               Likely pathogenic, Pathogenic
## 2     NA Pathogenic, Likely pathogenic, Pathogenic/Likely pathogenic
## 3     NA                               Likely pathogenic, Pathogenic
## 4     NA Pathogenic, Likely pathogenic, Pathogenic/Likely pathogenic
## 5     NA Pathogenic, Likely pathogenic, Pathogenic/Likely pathogenic
## 6     NA Pathogenic, Likely pathogenic, Pathogenic/Likely pathogenic
##         dbSNP somatic
## 1 rs121909224 Somatic
## 2 rs121909229 Somatic
## 3 rs121909224 Somatic
## 4 rs121909229 Somatic
## 5 rs121909229 Somatic
## 6 rs121909229 Somatic
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nrow}\NormalTok{(pten\_soma) }\CommentTok{\#2610}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2610
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(pten\_soma}\SpecialCharTok{$}\NormalTok{Annotation)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##                                                OncoKB: Inconclusive, level NA;CIViC: NA;MyCancerGenome: not present;CancerHotspot: no;3DHotspot: no 
##                                                                                                                                                   2 
##                                              OncoKB: Inconclusive, level NA;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes 
##                                                                                                                                                   6 
##                                              OncoKB: Likely Neutral, level NA;CIViC: NA;MyCancerGenome: not present;CancerHotspot: no;3DHotspot: no 
##                                                                                                                                                   2 
##                                    OncoKB: Likely Oncogenic, level_4;CIViC: Functional: 1;MyCancerGenome: present;CancerHotspot: yes;3DHotspot: yes 
##                                                                                                                                                 161 
##                                             OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: no;3DHotspot: no 
##                                                                                                                                                 309 
##                                            OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: no;3DHotspot: yes 
##                                                                                                                                                  83 
##                                            OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: no 
##                                                                                                                                                 358 
##                                           OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes 
##                                                                                                                                                 274 
##                                                 OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: present;CancerHotspot: no;3DHotspot: no 
##                                                                                                                                                   2 
##                                                OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: present;CancerHotspot: no;3DHotspot: yes 
##                                                                                                                                                  14 
##                                                OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: present;CancerHotspot: yes;3DHotspot: no 
##                                                                                                                                                  16 
##                                               OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: present;CancerHotspot: yes;3DHotspot: yes 
##                                                                                                                                                  11 
## OncoKB: Likely Oncogenic, level_4;CIViC: Oncogenic: 1, Diagnostic: 2, Predisposing: 2;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes 
##                                                                                                                                                  48 
##                                                 OncoKB: Oncogenic, level NA;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes 
##                                                                                                                                                   1 
##                                        OncoKB: Oncogenic, level_4;CIViC: Functional: 1;MyCancerGenome: not present;CancerHotspot: no;3DHotspot: yes 
##                                                                                                                                                  13 
##                                                    OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: no;3DHotspot: no 
##                                                                                                                                                  89 
##                                                   OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: no;3DHotspot: yes 
##                                                                                                                                                  53 
##                                                   OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: no 
##                                                                                                                                                  43 
##                                                  OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes 
##                                                                                                                                                 203 
##                                                      OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: present;CancerHotspot: yes;3DHotspot: yes 
##                                                                                                                                                 103 
##                                        OncoKB: Predicted Oncogenic, level NA;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: no 
##                                                                                                                                                   1 
##                                         OncoKB: Predicted Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: no 
##                                                                                                                                                  36 
##                                        OncoKB: Predicted Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes 
##                                                                                                                                                  98 
##                                                     OncoKB: Unknown, level NA;CIViC: NA;MyCancerGenome: not present;CancerHotspot: no;3DHotspot: no 
##                                                                                                                                                 588 
##                                                    OncoKB: Unknown, level NA;CIViC: NA;MyCancerGenome: not present;CancerHotspot: no;3DHotspot: yes 
##                                                                                                                                                  92 
##                                                   OncoKB: Unknown, level NA;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes 
##                                                                                                                                                   2 
##                                                         OncoKB: Unknown, level NA;CIViC: NA;MyCancerGenome: present;CancerHotspot: no;3DHotspot: no 
##                                                                                                                                                   2
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_soma\_hotspot }\OtherTok{\textless{}{-}}\NormalTok{ pten\_soma[}\FunctionTok{grepl}\NormalTok{(}\StringTok{"CancerHotspot: yes"}\NormalTok{, pten\_soma}\SpecialCharTok{$}\NormalTok{Annotation), ]}
\NormalTok{pten\_soma\_hotspot\_unique }\OtherTok{\textless{}{-}}\NormalTok{ pten\_soma\_hotspot }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{(Protein.Change, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_soma\_hotspot\_unique) }\CommentTok{\#144}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 144
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nrow}\NormalTok{(pten\_df) }\CommentTok{\#4873}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4873
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df\_clean }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{(mutant, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_df\_clean) }\CommentTok{\#4839}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4839
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(}\FunctionTok{intersect}\NormalTok{(pten\_soma\_hotspot}\SpecialCharTok{$}\NormalTok{Protein.Change, pten\_df\_clean}\SpecialCharTok{$}\NormalTok{mutant)) }\CommentTok{\#93}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 93
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(}\FunctionTok{setdiff}\NormalTok{(pten\_soma\_hotspot}\SpecialCharTok{$}\NormalTok{Protein.Change, pten\_df\_clean}\SpecialCharTok{$}\NormalTok{mutant)) }\CommentTok{\#51}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 51
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df\_clean}\SpecialCharTok{$}\NormalTok{somatic }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(}
\NormalTok{  pten\_df\_clean}\SpecialCharTok{$}\NormalTok{mutant }\SpecialCharTok{\%in\%}\NormalTok{ pten\_soma\_hotspot}\SpecialCharTok{$}\NormalTok{Protein.Change,}
  \StringTok{"Somatic"}\NormalTok{,}
  \ConstantTok{NA}
\NormalTok{)}
\FunctionTok{table}\NormalTok{(pten\_df\_clean}\SpecialCharTok{$}\NormalTok{somatic)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Somatic 
##      93
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_df, pten\_soma\_hotspot, }\AttributeTok{by.x=}\StringTok{"mutant"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"Protein.Change"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_df) }
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 6674
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{(mutant, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}

\FunctionTok{table}\NormalTok{(pten\_df}\SpecialCharTok{$}\NormalTok{somatic)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Somatic 
##      93
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_gnomad }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(HGVS.Consequence)) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{clinvar\_clinical\_significance }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"likely\_pathogenic"}\NormalTok{, }\StringTok{"pathogenic"}\NormalTok{, }\StringTok{"conflict"}\NormalTok{, }\StringTok{"likely\_risk\_allele"}\NormalTok{, }\StringTok{"uncertain\_risk\_allele"}\NormalTok{, }\StringTok{"VUS"}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{ClinVar.Germline.Classification }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"Pathogenic"}\NormalTok{, }\StringTok{"Pathogenic/Likely pathogenic"}\NormalTok{, }\StringTok{"Likely pathogenic"}\NormalTok{,}
                                                 \StringTok{"Conflicting classifications of pathogenicity"}\NormalTok{, }\StringTok{"Likely pathogenic/Likely risk allele"}\NormalTok{,}
                                                 \StringTok{"Likely risk allele"}\NormalTok{, }\StringTok{"Pathogenic/Likely pathogenic/Likely risk allele"}\NormalTok{,}
                                                 \StringTok{"Uncertain significance"}\NormalTok{, }\StringTok{"Uncertain significance/Uncertain risk allele"}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(Phenotype\_Class)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(somatic))}

\FunctionTok{nrow}\NormalTok{(pten\_gnomad) }\CommentTok{\#37}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(pten\_gnomad}\SpecialCharTok{$}\NormalTok{mutant)) }\CommentTok{\#37}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_gnomad}\SpecialCharTok{$}\NormalTok{var\_source }\OtherTok{\textless{}{-}} \StringTok{"gnomAD"}


\NormalTok{pten\_somatic }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_df, pten\_soma\_hotspot, }\AttributeTok{by.x=}\StringTok{"mutant"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"Protein.Change"}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_somatic) }\CommentTok{\#}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 788
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(pten\_somatic}\SpecialCharTok{$}\NormalTok{mutant)) }\CommentTok{\#93}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 93
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_somatic }\OtherTok{\textless{}{-}}\NormalTok{ pten\_somatic }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{(mutant, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{head}\NormalTok{(pten\_somatic)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   mutant DMS_score_abundance DMS_score_bin_abundance    ESM.1v
## 1  A126G           1.1031274                       1 -10.59961
## 2  A126S           1.1589273                       1 -13.37786
## 3  A126T           0.9698197                       1 -14.54880
## 4  C105G           0.3516079                       0 -13.94637
## 5  C105R           0.1588046                       0 -13.99142
## 6  C105S           0.2445279                       0 -10.05558
##   DMS_score_activity DMS_score_bin_activity mutation_position   V1      Model
## 1        -0.82914446                      1               126 2505 ThermoMPNN
## 2         0.06133791                      1               126 2515 ThermoMPNN
## 3        -1.93558841                      0               126 2516 ThermoMPNN
## 4        -4.83784503                      0               105 2085 ThermoMPNN
## 5        -3.78634247                      0               105 2094 ThermoMPNN
## 6        -1.05959009                      1               105 2095 ThermoMPNN
##                 Dataset  ddG_pred position wildtype mutation
## 1 AF-P60484-F1-model_v2 0.6510435      125        A        G
## 2 AF-P60484-F1-model_v2 0.2785859      125        A        S
## 3 AF-P60484-F1-model_v2 0.4760755      125        A        T
## 4 AF-P60484-F1-model_v2 1.0596524      104        C        G
## 5 AF-P60484-F1-model_v2 1.5254858      104        C        R
## 6 AF-P60484-F1-model_v2 0.4113229      104        C        S
##                     pdb chain new_position uniprot exposure_SS exposure_rASA
## 1 AF-P60484-F1-model_v2     A          126  P60484           S          0.33
## 2 AF-P60484-F1-model_v2     A          126  P60484           S          0.33
## 3 AF-P60484-F1-model_v2     A          126  P60484           S          0.33
## 4 AF-P60484-F1-model_v2     A          105  P60484           H          0.00
## 5 AF-P60484-F1-model_v2     A          105  P60484           H          0.00
## 6 AF-P60484-F1-model_v2     A          105  P60484           H          0.00
##   spot_disorder func_esms_residue_class func_esms_variant_class
## 1             0                       1                       1
## 2             0                       1                       1
## 3             0                       1                       1
## 4             0                       2                       2
## 5             0                       2                       2
## 6             0                       2                       1
##   clinvar_clinical_significance
## 1                           VUS
## 2                           VUS
## 3             likely_pathogenic
## 4             likely_pathogenic
## 5                              
## 6                              
##                                                                                                                                                                                                                                                                                                                                                                                                              sequence
## 1 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 2 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 3 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 4 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 5 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 6 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
##   Phenotype_Class Phenotype_List Fitness_Score Abundance_Score HGVS.Consequence
## 1            <NA>           <NA>            NA              NA             <NA>
## 2            <NA>           <NA>            NA              NA             <NA>
## 3            <NA>           <NA>            NA              NA             <NA>
## 4            <NA>           <NA>            NA              NA      p.Cys105Gly
## 5            <NA>           <NA>            NA              NA             <NA>
## 6            <NA>           <NA>            NA              NA             <NA>
##   ClinVar.Germline.Classification        rsIDs ClinVar.Variation.ID gnomad
## 1                            <NA>         <NA>                   NA   <NA>
## 2                            <NA>         <NA>                   NA   <NA>
## 3                            <NA>         <NA>                   NA   <NA>
## 4               Likely pathogenic rs1303165645              1728081 gnomad
## 5                            <NA>         <NA>                   NA   <NA>
## 6                            <NA>         <NA>                   NA   <NA>
##       fitted  residuals
## 1 -0.1823781 -0.6467664
## 2 -0.1908700  0.2522079
## 3 -0.1785532 -1.7570352
## 4 -1.3357357 -3.5021093
## 5 -2.3737286 -1.4126139
## 6 -1.9276639  0.8680738
##                                                        Study.x
## 1            Breast Invasive Carcinoma (TCGA, PanCancer Atlas)
## 2                        Breast Cancer (MSK, Cancer Cell 2018)
## 3 Uterine Corpus Endometrial Carcinoma (TCGA, PanCancer Atlas)
## 4                                  Melanomas (TCGA, Cell 2015)
## 5      Breast Cancer (METABRIC, Nature 2012 & Nat Commun 2016)
## 6  MSK-IMPACT Clinical Sequencing Cohort (MSKCC, Nat Med 2017)
##         Sample.ID.x                  Cancer.Type.x
## 1   TCGA-LL-A5YP-01                               
## 2 P-0000422-T02-IM3                               
## 3   TCGA-EY-A1GU-01                               
## 4   TCGA-D3-A2J8-06                               
## 5           MB-6143                               
## 6 P-0002672-T01-IM3 Gastrointestinal Stromal Tumor
##                                                                                                Annotation.x
## 1        OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
## 2        OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
## 3 OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
## 4 OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
## 5 OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
## 6 OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
##                                                                                                                         Functional.Impact.x
## 1       MutationAssessor: impact: low, score: 1.885;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.998
## 2     MutationAssessor: impact: medium, score: 3.39;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.998
## 3      MutationAssessor: impact: medium, score: 2.9;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.999
## 4 MutationAssessor: impact: medium, score: 3.465;SIFT: impact: deleterious, score: 0.01;Polyphen-2: impact: probably_damaging, score: 0.998
## 5       MutationAssessor: impact: high, score: 3.81;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.999
## 6     MutationAssessor: impact: medium, score: 2.53;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.998
##     Mutation.Type.x Variant.Type.x Copy...x COSMIC.x    MS.x    VS.x
## 1 Missense_Mutation            SNP                18       .       .
## 2 Missense_Mutation            SNP                18 UNKNOWN Unknown
## 3 Missense_Mutation            SNP                18       .       .
## 4 Missense_Mutation            SNP                21 Somatic    <NA>
## 5 Missense_Mutation            SNP                21    <NA>    <NA>
## 6 Missense_Mutation            SNP  diploid       21    <NA>    <NA>
##        Center.x Chromosome.x Start.Pos.x End.Pos.x Ref.x Var.x          HGVSg.x
## 1             .           10    89692893  89692893     C     G 10:g.89692893C>G
## 2         MSKCC           10    89692892  89692892     G     T 10:g.89692892G>T
## 3             .           10    89692892  89692892     G     A 10:g.89692892G>A
## 4 broad.mit.edu           10    89692829  89692829     T     G 10:g.89692829T>G
## 5      METABRIC           10    89692829  89692829     T     C 10:g.89692829T>C
## 6          <NA>           10    89692829  89692829     T     A 10:g.89692829T>A
##                      HGVSc.x Allele.Freq..T..x Allele.Freq..N..x
## 1 ENST00000371953.3:c.377C>G              0.50                NA
## 2 ENST00000371953.3:c.376G>T              0.48            0.0036
## 3 ENST00000371953.3:c.376G>A              0.46                NA
## 4 ENST00000371953.3:c.313T>G              0.16                NA
## 5 ENST00000371953.3:c.313T>C                NA                NA
## 6 ENST00000371953.3:c.313T>A              0.32                NA
##   Variant.Reads.x Ref.Reads.x Variant.Reads..Normal..x Ref.Reads..Normal..x
## 1              27          27                       NA                   82
## 2             105         115                        1                  275
## 3              43          51                       NA                   68
## 4              11          56                       NA                   NA
## 5              NA          NA                       NA                   NA
## 6              44          93                       NA                   NA
##   X..Mut.in.Sample.x Exon.x gnomAD.x                                 ClinVar.x
## 1                 64    5/9       NA                                          
## 2                  7    5/9       NA                    Uncertain significance
## 3               1179    5/9       NA Uncertain significance, Likely pathogenic
## 4                507    5/9       NA                                          
## 5                  6    5/9       NA                                          
## 6                 16    5/9       NA                                          
##        dbSNP.x somatic.x
## 1                Somatic
## 2 rs1554898129   Somatic
## 3 rs1554898129   Somatic
## 4                Somatic
## 5                Somatic
## 6                Somatic
##                                                        Study.y
## 1            Breast Invasive Carcinoma (TCGA, PanCancer Atlas)
## 2                        Breast Cancer (MSK, Cancer Cell 2018)
## 3 Uterine Corpus Endometrial Carcinoma (TCGA, PanCancer Atlas)
## 4                                  Melanomas (TCGA, Cell 2015)
## 5      Breast Cancer (METABRIC, Nature 2012 & Nat Commun 2016)
## 6  MSK-IMPACT Clinical Sequencing Cohort (MSKCC, Nat Med 2017)
##         Sample.ID.y                  Cancer.Type.y
## 1   TCGA-LL-A5YP-01                               
## 2 P-0000422-T02-IM3                               
## 3   TCGA-EY-A1GU-01                               
## 4   TCGA-D3-A2J8-06                               
## 5           MB-6143                               
## 6 P-0002672-T01-IM3 Gastrointestinal Stromal Tumor
##                                                                                                Annotation.y
## 1        OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
## 2        OncoKB: Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
## 3 OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
## 4 OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
## 5 OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
## 6 OncoKB: Likely Oncogenic, level_4;CIViC: NA;MyCancerGenome: not present;CancerHotspot: yes;3DHotspot: yes
##                                                                                                                         Functional.Impact.y
## 1       MutationAssessor: impact: low, score: 1.885;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.998
## 2     MutationAssessor: impact: medium, score: 3.39;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.998
## 3      MutationAssessor: impact: medium, score: 2.9;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.999
## 4 MutationAssessor: impact: medium, score: 3.465;SIFT: impact: deleterious, score: 0.01;Polyphen-2: impact: probably_damaging, score: 0.998
## 5       MutationAssessor: impact: high, score: 3.81;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.999
## 6     MutationAssessor: impact: medium, score: 2.53;SIFT: impact: deleterious, score: 0;Polyphen-2: impact: probably_damaging, score: 0.998
##     Mutation.Type.y Variant.Type.y Copy...y COSMIC.y    MS.y    VS.y
## 1 Missense_Mutation            SNP                18       .       .
## 2 Missense_Mutation            SNP                18 UNKNOWN Unknown
## 3 Missense_Mutation            SNP                18       .       .
## 4 Missense_Mutation            SNP                21 Somatic    <NA>
## 5 Missense_Mutation            SNP                21    <NA>    <NA>
## 6 Missense_Mutation            SNP  diploid       21    <NA>    <NA>
##        Center.y Chromosome.y Start.Pos.y End.Pos.y Ref.y Var.y          HGVSg.y
## 1             .           10    89692893  89692893     C     G 10:g.89692893C>G
## 2         MSKCC           10    89692892  89692892     G     T 10:g.89692892G>T
## 3             .           10    89692892  89692892     G     A 10:g.89692892G>A
## 4 broad.mit.edu           10    89692829  89692829     T     G 10:g.89692829T>G
## 5      METABRIC           10    89692829  89692829     T     C 10:g.89692829T>C
## 6          <NA>           10    89692829  89692829     T     A 10:g.89692829T>A
##                      HGVSc.y Allele.Freq..T..y Allele.Freq..N..y
## 1 ENST00000371953.3:c.377C>G              0.50                NA
## 2 ENST00000371953.3:c.376G>T              0.48            0.0036
## 3 ENST00000371953.3:c.376G>A              0.46                NA
## 4 ENST00000371953.3:c.313T>G              0.16                NA
## 5 ENST00000371953.3:c.313T>C                NA                NA
## 6 ENST00000371953.3:c.313T>A              0.32                NA
##   Variant.Reads.y Ref.Reads.y Variant.Reads..Normal..y Ref.Reads..Normal..y
## 1              27          27                       NA                   82
## 2             105         115                        1                  275
## 3              43          51                       NA                   68
## 4              11          56                       NA                   NA
## 5              NA          NA                       NA                   NA
## 6              44          93                       NA                   NA
##   X..Mut.in.Sample.y Exon.y gnomAD.y                                 ClinVar.y
## 1                 64    5/9       NA                                          
## 2                  7    5/9       NA                    Uncertain significance
## 3               1179    5/9       NA Uncertain significance, Likely pathogenic
## 4                507    5/9       NA                                          
## 5                  6    5/9       NA                                          
## 6                 16    5/9       NA                                          
##        dbSNP.y somatic.y
## 1                Somatic
## 2 rs1554898129   Somatic
## 3 rs1554898129   Somatic
## 4                Somatic
## 5                Somatic
## 6                Somatic
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_somatic }\OtherTok{\textless{}{-}}\NormalTok{ pten\_somatic }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{clinvar\_clinical\_significance }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"likely\_pathogenic"}\NormalTok{, }\StringTok{"pathogenic"}\NormalTok{, }\StringTok{"conflict"}\NormalTok{, }\StringTok{"likely\_risk\_allele"}\NormalTok{, }\StringTok{"uncertain\_risk\_allele"}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{ClinVar.Germline.Classification }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"Pathogenic"}\NormalTok{, }\StringTok{"Pathogenic/Likely pathogenic"}\NormalTok{, }\StringTok{"Likely pathogenic"}\NormalTok{,}
                                                 \StringTok{"Conflicting classifications of pathogenicity"}\NormalTok{, }\StringTok{"Likely pathogenic/Likely risk allele"}\NormalTok{,}
                                                 \StringTok{"Likely risk allele"}\NormalTok{, }\StringTok{"Pathogenic/Likely pathogenic/Likely risk allele"}\NormalTok{,}
                                                 \StringTok{"Uncertain significance"}\NormalTok{, }\StringTok{"Uncertain significance/Uncertain risk allele"}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(Phenotype\_Class))}
\FunctionTok{nrow}\NormalTok{(pten\_somatic) }\CommentTok{\#37}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_somatic}\SpecialCharTok{$}\NormalTok{var\_source }\OtherTok{\textless{}{-}} \StringTok{"Somatic"}

\FunctionTok{colnames}\NormalTok{(pten\_somatic)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "mutant"                          "DMS_score_abundance"            
##  [3] "DMS_score_bin_abundance"         "ESM.1v"                         
##  [5] "DMS_score_activity"              "DMS_score_bin_activity"         
##  [7] "mutation_position"               "V1"                             
##  [9] "Model"                           "Dataset"                        
## [11] "ddG_pred"                        "position"                       
## [13] "wildtype"                        "mutation"                       
## [15] "pdb"                             "chain"                          
## [17] "new_position"                    "uniprot"                        
## [19] "exposure_SS"                     "exposure_rASA"                  
## [21] "spot_disorder"                   "func_esms_residue_class"        
## [23] "func_esms_variant_class"         "clinvar_clinical_significance"  
## [25] "sequence"                        "Phenotype_Class"                
## [27] "Phenotype_List"                  "Fitness_Score"                  
## [29] "Abundance_Score"                 "HGVS.Consequence"               
## [31] "ClinVar.Germline.Classification" "rsIDs"                          
## [33] "ClinVar.Variation.ID"            "gnomad"                         
## [35] "fitted"                          "residuals"                      
## [37] "Study.x"                         "Sample.ID.x"                    
## [39] "Cancer.Type.x"                   "Annotation.x"                   
## [41] "Functional.Impact.x"             "Mutation.Type.x"                
## [43] "Variant.Type.x"                  "Copy...x"                       
## [45] "COSMIC.x"                        "MS.x"                           
## [47] "VS.x"                            "Center.x"                       
## [49] "Chromosome.x"                    "Start.Pos.x"                    
## [51] "End.Pos.x"                       "Ref.x"                          
## [53] "Var.x"                           "HGVSg.x"                        
## [55] "HGVSc.x"                         "Allele.Freq..T..x"              
## [57] "Allele.Freq..N..x"               "Variant.Reads.x"                
## [59] "Ref.Reads.x"                     "Variant.Reads..Normal..x"       
## [61] "Ref.Reads..Normal..x"            "X..Mut.in.Sample.x"             
## [63] "Exon.x"                          "gnomAD.x"                       
## [65] "ClinVar.x"                       "dbSNP.x"                        
## [67] "somatic.x"                       "Study.y"                        
## [69] "Sample.ID.y"                     "Cancer.Type.y"                  
## [71] "Annotation.y"                    "Functional.Impact.y"            
## [73] "Mutation.Type.y"                 "Variant.Type.y"                 
## [75] "Copy...y"                        "COSMIC.y"                       
## [77] "MS.y"                            "VS.y"                           
## [79] "Center.y"                        "Chromosome.y"                   
## [81] "Start.Pos.y"                     "End.Pos.y"                      
## [83] "Ref.y"                           "Var.y"                          
## [85] "HGVSg.y"                         "HGVSc.y"                        
## [87] "Allele.Freq..T..y"               "Allele.Freq..N..y"              
## [89] "Variant.Reads.y"                 "Ref.Reads.y"                    
## [91] "Variant.Reads..Normal..y"        "Ref.Reads..Normal..y"           
## [93] "X..Mut.in.Sample.y"              "Exon.y"                         
## [95] "gnomAD.y"                        "ClinVar.y"                      
## [97] "dbSNP.y"                         "somatic.y"                      
## [99] "var_source"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_patho }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(clinvar\_clinical\_significance }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"likely\_pathogenic"}\NormalTok{, }\StringTok{"pathogenic"}\NormalTok{))}
\FunctionTok{nrow}\NormalTok{(pten\_patho) }\CommentTok{\#}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 115
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(pten\_patho}\SpecialCharTok{$}\NormalTok{mutant)) }\CommentTok{\#115 }
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 115
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_patho }\OtherTok{\textless{}{-}}\NormalTok{ pten\_patho }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{(mutant, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_patho) }\CommentTok{\#115}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 115
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_patho}\SpecialCharTok{$}\NormalTok{var\_source }\OtherTok{\textless{}{-}} \StringTok{"Pathogenic"}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p1 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(pten\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity, }\AttributeTok{color =}\NormalTok{ residuals)) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.35}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_vline}\NormalTok{(}\AttributeTok{xintercept =} \FloatTok{0.71}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{linewidth =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{color =} \StringTok{"grey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_hline}\NormalTok{(}\AttributeTok{yintercept =} \SpecialCharTok{{-}}\FloatTok{1.11}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{linewidth =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{color =} \StringTok{"grey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_line}\NormalTok{(}\AttributeTok{data =}\NormalTok{ fit\_line\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity),}
            \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{linewidth =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{geom\_text\_repel}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df, mutant }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"R130G"}\NormalTok{, }\StringTok{"C124R"}\NormalTok{)),}
                  \FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ mutant),}
                  \AttributeTok{size =} \DecValTok{4}\NormalTok{,}\AttributeTok{color =} \StringTok{"black"}\NormalTok{,}
                  \AttributeTok{max.overlaps =} \ConstantTok{Inf}\NormalTok{, }\AttributeTok{box.padding =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{point.padding =} \FloatTok{0.3}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df, mutant }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"R130G"}\NormalTok{, }\StringTok{"C124R"}\NormalTok{)),}
             \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity),}
             \AttributeTok{shape =} \DecValTok{17}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{size =} \DecValTok{3}\NormalTok{) }\SpecialCharTok{+}  \CommentTok{\# Triangle shape}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"All 4839 Variants"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{"Measured abundance"}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Measured activity"}\NormalTok{,}
    \AttributeTok{color =} \StringTok{"Residuals"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{ylim}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{5.8}\NormalTok{, }\FloatTok{2.9}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{xlim}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{0.3}\NormalTok{, }\FloatTok{1.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_color\_viridis}\NormalTok{(}\AttributeTok{option =} \StringTok{"C"}\NormalTok{, }\AttributeTok{direction =} \DecValTok{1}\NormalTok{, }\AttributeTok{limits =} \FunctionTok{c}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{4.8}\NormalTok{, }\FloatTok{5.2}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}

\CommentTok{\# Add marginal density plots}
\NormalTok{p1 }\OtherTok{\textless{}{-}} \FunctionTok{ggMarginal}\NormalTok{(}
\NormalTok{  p1,}
  \AttributeTok{type =} \StringTok{"density"}\NormalTok{,}
  \AttributeTok{margins =} \StringTok{"both"}\NormalTok{,}
  \AttributeTok{groupColour =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{groupFill =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{size =} \DecValTok{10}\NormalTok{,}
  \AttributeTok{colour =} \StringTok{"grey"}\NormalTok{,}
  \AttributeTok{fill =} \StringTok{"lightgrey"}
\NormalTok{)}

\FunctionTok{nrow}\NormalTok{(pten\_gnomad)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p2 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(pten\_gnomad, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity, }\AttributeTok{color =}\NormalTok{ residuals)) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{size =} \DecValTok{2}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_vline}\NormalTok{(}\AttributeTok{xintercept =} \FloatTok{0.71}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{linewidth =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{color =} \StringTok{"grey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_hline}\NormalTok{(}\AttributeTok{yintercept =} \SpecialCharTok{{-}}\FloatTok{1.11}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{linewidth =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{color =} \StringTok{"grey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_line}\NormalTok{(}\AttributeTok{data =}\NormalTok{ fit\_line\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity),}
            \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{linewidth =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"gnomAD Variants"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{"Abundance"}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Activity"}\NormalTok{,}
    \AttributeTok{color =} \StringTok{"Loess residuals"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{ylim}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{5.8}\NormalTok{, }\FloatTok{2.9}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{xlim}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{0.3}\NormalTok{, }\FloatTok{1.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_color\_viridis}\NormalTok{(}\AttributeTok{option =} \StringTok{"C"}\NormalTok{, }\AttributeTok{direction =} \DecValTok{1}\NormalTok{, }\AttributeTok{limits =} \FunctionTok{c}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{4.8}\NormalTok{, }\FloatTok{5.2}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{geom\_text\_repel}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df, mutant }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"A79T"}\NormalTok{)),}
                  \FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ mutant),}
                  \AttributeTok{size =} \DecValTok{4}\NormalTok{,}\AttributeTok{color =} \StringTok{"black"}\NormalTok{,}
                  \AttributeTok{max.overlaps =} \ConstantTok{Inf}\NormalTok{, }\AttributeTok{box.padding =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{point.padding =} \FloatTok{0.3}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df, mutant }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"A79T"}\NormalTok{)),}
             \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity),}
             \AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{) }

\CommentTok{\# Add marginal density plots}
\NormalTok{p2 }\OtherTok{\textless{}{-}} \FunctionTok{ggMarginal}\NormalTok{(}
\NormalTok{  p2,}
  \AttributeTok{type =} \StringTok{"density"}\NormalTok{,}
  \AttributeTok{margins =} \StringTok{"both"}\NormalTok{,}
  \AttributeTok{groupColour =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{groupFill =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{size =} \DecValTok{10}\NormalTok{,}
  \AttributeTok{colour =} \StringTok{"grey"}\NormalTok{,}
  \AttributeTok{fill =} \StringTok{"lightgrey"}
\NormalTok{)}


\FunctionTok{nrow}\NormalTok{(pten\_patho)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 115
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p3 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(pten\_patho, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity, }\AttributeTok{color =}\NormalTok{ residuals)) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{shape =} \DecValTok{17}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_vline}\NormalTok{(}\AttributeTok{xintercept =} \FloatTok{0.71}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{linewidth =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{color =} \StringTok{"grey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_hline}\NormalTok{(}\AttributeTok{yintercept =} \SpecialCharTok{{-}}\FloatTok{1.11}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{linewidth =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{color =} \StringTok{"grey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_line}\NormalTok{(}\AttributeTok{data =}\NormalTok{ fit\_line\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity),}
            \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{linewidth =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
      \FunctionTok{geom\_text\_repel}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df, mutant }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"R130G"}\NormalTok{, }\StringTok{"C124R"}\NormalTok{)),}
                  \FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ mutant),}
                  \AttributeTok{size =} \DecValTok{4}\NormalTok{,}\AttributeTok{color =} \StringTok{"black"}\NormalTok{,}
                  \AttributeTok{max.overlaps =} \ConstantTok{Inf}\NormalTok{, }\AttributeTok{box.padding =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{point.padding =} \FloatTok{0.3}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df, mutant }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"R130G"}\NormalTok{, }\StringTok{"C124R"}\NormalTok{)),}
             \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity),}
             \AttributeTok{shape =} \DecValTok{17}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{size =} \DecValTok{3}\NormalTok{) }\SpecialCharTok{+}  \CommentTok{\# Triangle shape}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"ClinVar Pathogenic Variants"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{"Abundance"}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Activity"}\NormalTok{,}
    \AttributeTok{color =} \StringTok{"Loess residuals"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{ylim}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{5.8}\NormalTok{, }\FloatTok{2.9}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{xlim}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{0.3}\NormalTok{, }\FloatTok{1.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_color\_viridis}\NormalTok{(}\AttributeTok{option =} \StringTok{"C"}\NormalTok{, }\AttributeTok{direction =} \DecValTok{1}\NormalTok{, }\AttributeTok{limits =} \FunctionTok{c}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{4.8}\NormalTok{, }\FloatTok{5.2}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}

\CommentTok{\# Add marginal density plots}
\NormalTok{p3 }\OtherTok{\textless{}{-}} \FunctionTok{ggMarginal}\NormalTok{(}
\NormalTok{  p3,}
  \AttributeTok{type =} \StringTok{"density"}\NormalTok{,}
  \AttributeTok{margins =} \StringTok{"both"}\NormalTok{,}
  \AttributeTok{groupColour =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{groupFill =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{size =} \DecValTok{10}\NormalTok{,}
  \AttributeTok{colour =} \StringTok{"grey"}\NormalTok{,}
  \AttributeTok{fill =} \StringTok{"lightgrey"}
\NormalTok{)}

\NormalTok{p4 }\OtherTok{\textless{}{-}} \FunctionTok{plot\_grid}\NormalTok{(p1,p2,p3,}\AttributeTok{ncol=}\DecValTok{3}\NormalTok{, }\AttributeTok{nrow=}\DecValTok{1}\NormalTok{)}

\FunctionTok{ggsave}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/figs/panels/fig4\_scatter1.pdf"}\NormalTok{, }
       \AttributeTok{plot =}\NormalTok{ p4, }\AttributeTok{width =} \DecValTok{9}\NormalTok{, }\AttributeTok{height =} \DecValTok{3}\NormalTok{, }\AttributeTok{dpi =} \DecValTok{300}\NormalTok{)}
\NormalTok{p4}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-5-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggplot}\NormalTok{(pten\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity, }\AttributeTok{color =}\NormalTok{ residuals)) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.35}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_vline}\NormalTok{(}\AttributeTok{xintercept =} \FloatTok{0.71}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{linewidth =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{color =} \StringTok{"grey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_hline}\NormalTok{(}\AttributeTok{yintercept =} \SpecialCharTok{{-}}\FloatTok{1.11}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{linewidth =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{color =} \StringTok{"grey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_line}\NormalTok{(}\AttributeTok{data =}\NormalTok{ fit\_line\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity),}
            \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{linewidth =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{geom\_text\_repel}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df, mutant }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"R130G"}\NormalTok{, }\StringTok{"C124R"}\NormalTok{)),}
                  \FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ mutant),}
                  \AttributeTok{size =} \DecValTok{4}\NormalTok{,}\AttributeTok{color =} \StringTok{"black"}\NormalTok{,}
                  \AttributeTok{max.overlaps =} \ConstantTok{Inf}\NormalTok{, }\AttributeTok{box.padding =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{point.padding =} \FloatTok{0.3}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df, mutant }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"R130G"}\NormalTok{, }\StringTok{"C124R"}\NormalTok{)),}
             \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ DMS\_score\_abundance, }\AttributeTok{y =}\NormalTok{ DMS\_score\_activity),}
             \AttributeTok{shape =} \DecValTok{17}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{size =} \DecValTok{3}\NormalTok{) }\SpecialCharTok{+}  \CommentTok{\# Triangle shape}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"All 4839 Variants"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{"Measured abundance"}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Measured activity"}\NormalTok{,}
    \AttributeTok{color =} \StringTok{"Residuals"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{ylim}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{5.8}\NormalTok{, }\FloatTok{2.9}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{xlim}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{0.3}\NormalTok{, }\FloatTok{1.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_color\_viridis}\NormalTok{(}\AttributeTok{option =} \StringTok{"C"}\NormalTok{, }\AttributeTok{direction =} \DecValTok{1}\NormalTok{, }\AttributeTok{limits =} \FunctionTok{c}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{4.8}\NormalTok{, }\FloatTok{5.2}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"bottom"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-6-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# 1. Set up}
\FunctionTok{set.seed}\NormalTok{(}\DecValTok{123}\NormalTok{)}
\NormalTok{n\_boot }\OtherTok{\textless{}{-}} \DecValTok{1000}
\NormalTok{match\_window }\OtherTok{\textless{}{-}} \FloatTok{0.05}
\NormalTok{n\_patho }\OtherTok{\textless{}{-}} \FunctionTok{nrow}\NormalTok{(pten\_patho)}

\CommentTok{\# 2. Get abundance/residuals from pten\_patho}
\NormalTok{patho\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_patho }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(DMS\_score\_abundance, residuals)}
\NormalTok{patho\_df}\SpecialCharTok{$}\NormalTok{group }\OtherTok{\textless{}{-}} \StringTok{"Pathogenic"}

\CommentTok{\# 3. Bootstrap sampling from non{-}patho pool with abundance matching}
\NormalTok{non\_patho\_pool }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{(mutant }\SpecialCharTok{\%in\%}\NormalTok{ pten\_patho}\SpecialCharTok{$}\NormalTok{mutant)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{(mutant }\SpecialCharTok{\%in\%}\NormalTok{ pten\_somatic}\SpecialCharTok{$}\NormalTok{mutant)) }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(DMS\_score\_abundance, residuals)}

\NormalTok{bootstrap\_medians }\OtherTok{\textless{}{-}} \FunctionTok{vector}\NormalTok{(}\StringTok{"numeric"}\NormalTok{, }\AttributeTok{length =}\NormalTok{ n\_boot)}

\CommentTok{\# Pre{-}group non{-}patho pool into bins}
\NormalTok{non\_patho\_pool }\OtherTok{\textless{}{-}}\NormalTok{ non\_patho\_pool }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{bin =} \FunctionTok{cut}\NormalTok{(DMS\_score\_abundance, }\AttributeTok{breaks =} \FunctionTok{seq}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{0.5}\NormalTok{, }\DecValTok{2}\NormalTok{, }\AttributeTok{by =}\NormalTok{ match\_window)))}

\CommentTok{\# Bin pathogenic variants accordingly}
\NormalTok{patho\_df }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{bin =} \FunctionTok{cut}\NormalTok{(DMS\_score\_abundance, }\AttributeTok{breaks =} \FunctionTok{seq}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{0.5}\NormalTok{, }\DecValTok{2}\NormalTok{, }\AttributeTok{by =}\NormalTok{ match\_window)))}

\CommentTok{\# Create lookup table for fast sampling}
\NormalTok{bin\_lookup }\OtherTok{\textless{}{-}} \FunctionTok{split}\NormalTok{(non\_patho\_pool}\SpecialCharTok{$}\NormalTok{residuals, non\_patho\_pool}\SpecialCharTok{$}\NormalTok{bin)}

\CommentTok{\# Bootstrap matrix}
\NormalTok{bootstrap\_matrix }\OtherTok{\textless{}{-}} \FunctionTok{matrix}\NormalTok{(}\ConstantTok{NA}\NormalTok{, }\AttributeTok{nrow =}\NormalTok{ n\_boot, }\AttributeTok{ncol =}\NormalTok{ n\_patho)}

\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\NormalTok{n\_boot) \{}
  \ControlFlowTok{for}\NormalTok{ (j }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\NormalTok{n\_patho) \{}
\NormalTok{    bin\_j }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df}\SpecialCharTok{$}\NormalTok{bin[j]}
\NormalTok{    candidates }\OtherTok{\textless{}{-}}\NormalTok{ bin\_lookup[[}\FunctionTok{as.character}\NormalTok{(bin\_j)]]}
    \ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{is.null}\NormalTok{(candidates) }\SpecialCharTok{\&\&} \FunctionTok{length}\NormalTok{(candidates) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) \{}
\NormalTok{      bootstrap\_matrix[i, j] }\OtherTok{\textless{}{-}} \FunctionTok{sample}\NormalTok{(candidates, }\DecValTok{1}\NormalTok{)}
\NormalTok{    \}}
\NormalTok{  \}}
\NormalTok{\}}

\CommentTok{\# Summarize into a dataframe}
\NormalTok{boot\_df }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{group =} \StringTok{"Random abundance{-}matched"}\NormalTok{,}
  \AttributeTok{residuals =} \FunctionTok{apply}\NormalTok{(bootstrap\_matrix, }\DecValTok{1}\NormalTok{, median, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{)}

\CommentTok{\# Combine with patho residuals}
\NormalTok{plot\_df }\OtherTok{\textless{}{-}} \FunctionTok{bind\_rows}\NormalTok{(}
\NormalTok{  patho\_df }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(group, residuals),}
\NormalTok{  boot\_df}
\NormalTok{)}
\NormalTok{label\_df }\OtherTok{\textless{}{-}}\NormalTok{ plot\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(group) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{(),}
    \AttributeTok{median\_val =} \FunctionTok{median}\NormalTok{(residuals),}
    \AttributeTok{y\_max =} \FunctionTok{max}\NormalTok{(residuals),}
    \AttributeTok{.groups =} \StringTok{"drop"}
\NormalTok{  )}

\NormalTok{label\_df }\OtherTok{\textless{}{-}}\NormalTok{ label\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{n\_label =} \FunctionTok{case\_when}\NormalTok{(}
\NormalTok{    group }\SpecialCharTok{==} \StringTok{"Random abundance{-}matched"} \SpecialCharTok{\textasciitilde{}} \StringTok{"bootstrapped 1000 times"}\NormalTok{,}
    \ConstantTok{TRUE} \SpecialCharTok{\textasciitilde{}} \FunctionTok{paste0}\NormalTok{(}\StringTok{"n = "}\NormalTok{, n)}
\NormalTok{  ))}

\CommentTok{\# Plot}
\NormalTok{p\_fast }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(plot\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ group, }\AttributeTok{y =}\NormalTok{ residuals, }\AttributeTok{fill =}\NormalTok{ group)) }\SpecialCharTok{+}
  \FunctionTok{geom\_violin}\NormalTok{(}\AttributeTok{trim =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{scale =} \StringTok{"width"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{color =} \ConstantTok{NA}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.15}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.7}\NormalTok{, }\AttributeTok{color =} \StringTok{"lightgrey"}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"crossbar"}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{fatten =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"point"}\NormalTok{, }\AttributeTok{shape =} \DecValTok{23}\NormalTok{, }\AttributeTok{size =} \FloatTok{2.5}\NormalTok{,}
               \AttributeTok{fill =} \StringTok{"black"}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}

 \FunctionTok{geom\_text}\NormalTok{(}
  \AttributeTok{data =}\NormalTok{ label\_df,}
  \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ group, }\AttributeTok{y =} \FloatTok{4.5}\NormalTok{, }\AttributeTok{label =}\NormalTok{ n\_label),}
  \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{size =} \DecValTok{4}\NormalTok{) }\SpecialCharTok{+}

  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ label\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ group, }\AttributeTok{y =}\NormalTok{ median\_val }\SpecialCharTok{+} \FloatTok{0.25}\NormalTok{, }\AttributeTok{label =} \FunctionTok{sprintf}\NormalTok{(}\StringTok{" \%.2f"}\NormalTok{, median\_val)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{6}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{x =} \StringTok{""}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Activity{-}abundance residuals"}\NormalTok{,}
    \AttributeTok{title =} \StringTok{""}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{(}\AttributeTok{base\_size =} \DecValTok{14}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"Pathogenic"} \OtherTok{=} \StringTok{"cornflowerblue"}\NormalTok{, }\StringTok{"Random abundance{-}matched"} \OtherTok{=} \StringTok{"darkolivegreen3"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_signif}\NormalTok{(}\AttributeTok{comparisons =} \FunctionTok{list}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"Pathogenic"}\NormalTok{, }\StringTok{"Random abundance{-}matched"}\NormalTok{)),}
              \AttributeTok{map\_signif\_level =} \ConstantTok{FALSE}\NormalTok{,}
              \AttributeTok{test =} \StringTok{"wilcox.test"}\NormalTok{,}
              \AttributeTok{tip\_length =} \FloatTok{0.01}\NormalTok{)}

\FunctionTok{ggsave}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/figs/panels/fig4\_violin1.pdf"}\NormalTok{, }
       \AttributeTok{plot =}\NormalTok{ p\_fast, }\AttributeTok{width =} \DecValTok{3}\NormalTok{, }\AttributeTok{height =} \DecValTok{4}\NormalTok{, }\AttributeTok{dpi =} \DecValTok{300}\NormalTok{)}
\NormalTok{p\_fast}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-7-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{patho\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{(mutant, Phenotype\_Class, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(patho\_df) }\CommentTok{\#4839}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4839
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{patho\_df1 }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(Phenotype\_Class }\SpecialCharTok{==} \StringTok{"ASD/DD"}\NormalTok{)}
\NormalTok{patho\_df2 }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(Phenotype\_Class }\SpecialCharTok{==} \StringTok{"ASD/DD \& PHTS"}\NormalTok{)}
\NormalTok{patho\_df3 }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(Phenotype\_Class }\SpecialCharTok{==} \StringTok{"PHTS"}\NormalTok{)}

\NormalTok{patho\_df1}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \StringTok{"ASD/DD"}
\NormalTok{patho\_df2}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \StringTok{"ASD/DD \& PHTS"}
\NormalTok{patho\_df3}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \StringTok{"PHTS"}
\NormalTok{pten\_gnomad}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \StringTok{"gnomAD"}
\NormalTok{pten\_somatic}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \StringTok{"somatic"}
\FunctionTok{colnames}\NormalTok{(patho\_df1)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "mutant"                          "DMS_score_abundance"            
##  [3] "DMS_score_bin_abundance"         "ESM.1v"                         
##  [5] "DMS_score_activity"              "DMS_score_bin_activity"         
##  [7] "mutation_position"               "V1"                             
##  [9] "Model"                           "Dataset"                        
## [11] "ddG_pred"                        "position"                       
## [13] "wildtype"                        "mutation"                       
## [15] "pdb"                             "chain"                          
## [17] "new_position"                    "uniprot"                        
## [19] "exposure_SS"                     "exposure_rASA"                  
## [21] "spot_disorder"                   "func_esms_residue_class"        
## [23] "func_esms_variant_class"         "clinvar_clinical_significance"  
## [25] "sequence"                        "Phenotype_Class"                
## [27] "Phenotype_List"                  "Fitness_Score"                  
## [29] "Abundance_Score"                 "HGVS.Consequence"               
## [31] "ClinVar.Germline.Classification" "rsIDs"                          
## [33] "ClinVar.Variation.ID"            "gnomad"                         
## [35] "fitted"                          "residuals"                      
## [37] "somatic"                         "var_class2"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{patho\_df1 }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df1 }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, DMS\_score\_abundance, DMS\_score\_activity, var\_class2, residuals)}
\NormalTok{patho\_df2 }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df2 }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, DMS\_score\_abundance, DMS\_score\_activity, var\_class2,residuals)}
\NormalTok{patho\_df3 }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df3 }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, DMS\_score\_abundance, DMS\_score\_activity, var\_class2,residuals)}
\NormalTok{pten\_gnomad }\OtherTok{\textless{}{-}}\NormalTok{ pten\_gnomad }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, DMS\_score\_abundance, DMS\_score\_activity, var\_class2,residuals)}
\NormalTok{pten\_somatic }\OtherTok{\textless{}{-}}\NormalTok{ pten\_somatic }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, DMS\_score\_abundance, DMS\_score\_activity, var\_class2,residuals)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{combined\_df }\OtherTok{\textless{}{-}} \FunctionTok{rbind}\NormalTok{(patho\_df1,patho\_df2,patho\_df3,pten\_gnomad, pten\_somatic)}
\NormalTok{median\_df }\OtherTok{\textless{}{-}}\NormalTok{ combined\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(var\_class2) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{median\_residual =} \FunctionTok{median}\NormalTok{(residuals),}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{()}
\NormalTok{  )}
\NormalTok{median\_df}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 5 x 3
##   var_class2    median_residual     n
##   <chr>                   <dbl> <int>
## 1 ASD/DD                0.162      12
## 2 ASD/DD & PHTS        -1.48       10
## 3 PHTS                 -1.77       36
## 4 gnomAD               -0.00905    37
## 5 somatic              -1.52       37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#wilcox.test(patho\_df1$residuals, pten\_gnomad$residuals)}
\NormalTok{combined\_df}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \FunctionTok{factor}\NormalTok{(}
\NormalTok{  combined\_df}\SpecialCharTok{$}\NormalTok{var\_class2,}
  \AttributeTok{levels =} \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"somatic"}\NormalTok{,}\StringTok{"ASD/DD"}\NormalTok{, }\StringTok{"ASD/DD \& PHTS"}\NormalTok{, }\StringTok{"PHTS"}\NormalTok{)}
\NormalTok{)}

\NormalTok{custom\_colors }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \StringTok{"gnomAD"} \OtherTok{=} \StringTok{"\#1b9e77"}\NormalTok{,   }\CommentTok{\# Teal green}
  \StringTok{"somatic"}   \OtherTok{=} \StringTok{"\#fbb4ae"}\NormalTok{,}
  \StringTok{"ASD/DD"}   \OtherTok{=} \StringTok{"\#f4a6b3"}\NormalTok{,   }\CommentTok{\# Warm orange}
  \StringTok{"ASD/DD \& PHTS"}   \OtherTok{=} \StringTok{"\#e7298a"}\NormalTok{,   }\CommentTok{\# Muted purple}
  \StringTok{"PHTS"}   \OtherTok{=} \StringTok{"\#7570b3"}    \CommentTok{\# Hot pink / magenta}
\NormalTok{)}

\NormalTok{highlight\_muts }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"C124S"}\NormalTok{, }\StringTok{"G129E"}\NormalTok{, }\StringTok{"R130G"}\NormalTok{, }\StringTok{"R130Q"}\NormalTok{, }\StringTok{"P38S"}\NormalTok{, }\StringTok{"R130P"}\NormalTok{, }\StringTok{"D92H"}\NormalTok{)}

\NormalTok{p10 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(combined\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_class2, }\AttributeTok{y =}\NormalTok{ residuals, }\AttributeTok{fill =}\NormalTok{ var\_class2)) }\SpecialCharTok{+}
  \FunctionTok{geom\_violin}\NormalTok{(}\AttributeTok{trim =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{scale =} \StringTok{"width"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{color =} \ConstantTok{NA}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.15}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.7}\NormalTok{, }\AttributeTok{color =} \StringTok{"lightgrey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"crossbar"}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{fatten =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"point"}\NormalTok{, }\AttributeTok{shape =} \DecValTok{23}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{fill =} \StringTok{"black"}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
  \AttributeTok{data =}\NormalTok{ median\_df,}
  \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_class2, }\AttributeTok{y =}\NormalTok{ median\_residual }\SpecialCharTok{+} \FloatTok{0.5}\NormalTok{, }\AttributeTok{label =} \FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%.2f"}\NormalTok{, median\_residual)),}
  \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{size =} \DecValTok{5}
\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =}\NormalTok{ custom\_colors) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_class2, }\AttributeTok{y =} \FloatTok{7.5}\NormalTok{, }\AttributeTok{label =} \FunctionTok{paste0}\NormalTok{(}\StringTok{"n = "}\NormalTok{, n)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \FloatTok{4.5}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"Across groups"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{""}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Activity{-}abundance residual"}\NormalTok{,}
    \AttributeTok{fill =} \StringTok{""}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{(}\AttributeTok{base\_size =} \DecValTok{14}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{,}
    \AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle =} \DecValTok{20}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{) }
\NormalTok{  ) }\SpecialCharTok{+}
    \FunctionTok{geom\_signif}\NormalTok{(}
    \AttributeTok{comparisons =} \FunctionTok{list}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"somatic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"ASD/DD"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"ASD/DD \& PHTS"}\NormalTok{),}
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"PHTS"}\NormalTok{)),}
    \AttributeTok{map\_signif\_level =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{test =} \StringTok{"wilcox.test"}\NormalTok{,}
    \AttributeTok{step\_increase =} \FloatTok{0.1}\NormalTok{,}
    \AttributeTok{textsize =} \DecValTok{3}
\NormalTok{  ) }

\FunctionTok{ggsave}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/figs/panels/fig4\_violin2.pdf"}\NormalTok{, }
       \AttributeTok{plot =}\NormalTok{ p10, }\AttributeTok{width =} \DecValTok{3}\NormalTok{, }\AttributeTok{height =} \DecValTok{4}\NormalTok{, }\AttributeTok{dpi =} \DecValTok{300}\NormalTok{)}
\NormalTok{p10}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-9-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wilcox.test}\NormalTok{(patho\_df1}\SpecialCharTok{$}\NormalTok{residuals, pten\_gnomad}\SpecialCharTok{$}\NormalTok{residuals) }\CommentTok{\#0.688}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum exact test
## 
## data:  patho_df1$residuals and pten_gnomad$residuals
## W = 235, p-value = 0.7743
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wilcox.test}\NormalTok{(patho\_df2}\SpecialCharTok{$}\NormalTok{residuals, pten\_gnomad}\SpecialCharTok{$}\NormalTok{residuals) }\CommentTok{\#0.03351}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum exact test
## 
## data:  patho_df2$residuals and pten_gnomad$residuals
## W = 82, p-value = 0.006283
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wilcox.test}\NormalTok{(patho\_df3}\SpecialCharTok{$}\NormalTok{residuals, pten\_gnomad}\SpecialCharTok{$}\NormalTok{residuals) }\CommentTok{\#9.383e{-}05}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum exact test
## 
## data:  patho_df3$residuals and pten_gnomad$residuals
## W = 299, p-value = 3.013e-05
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wilcox.test}\NormalTok{(patho\_df1}\SpecialCharTok{$}\NormalTok{residuals, patho\_df2}\SpecialCharTok{$}\NormalTok{residuals) }\CommentTok{\#0.1693}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum exact test
## 
## data:  patho_df1$residuals and patho_df2$residuals
## W = 93, p-value = 0.02996
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wilcox.test}\NormalTok{(patho\_df1}\SpecialCharTok{$}\NormalTok{residuals, patho\_df3}\SpecialCharTok{$}\NormalTok{residuals) }\CommentTok{\#0.02058}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum exact test
## 
## data:  patho_df1$residuals and patho_df3$residuals
## W = 336, p-value = 0.003475
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wilcox.test}\NormalTok{(patho\_df2}\SpecialCharTok{$}\NormalTok{residuals, patho\_df3}\SpecialCharTok{$}\NormalTok{residuals) }\CommentTok{\#0.3959}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum exact test
## 
## data:  patho_df2$residuals and patho_df3$residuals
## W = 202, p-value = 0.5725
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nrow}\NormalTok{(pten\_gnomad) }\CommentTok{\#37}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nrow}\NormalTok{(pten\_patho) }\CommentTok{\#115}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 115
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nrow}\NormalTok{(pten\_somatic) }\CommentTok{\#37}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_gnomad}\SpecialCharTok{$}\NormalTok{var\_source }\OtherTok{\textless{}{-}} \StringTok{"gnomAD"}
\NormalTok{pten\_patho}\SpecialCharTok{$}\NormalTok{var\_source }\OtherTok{\textless{}{-}} \StringTok{"Pathogenic"}
\NormalTok{pten\_somatic}\SpecialCharTok{$}\NormalTok{var\_source }\OtherTok{\textless{}{-}} \StringTok{"Somatic"}

\NormalTok{pten\_gnomad }\OtherTok{\textless{}{-}}\NormalTok{ pten\_gnomad }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, residuals, var\_source,DMS\_score\_abundance)}
\NormalTok{pten\_patho }\OtherTok{\textless{}{-}}\NormalTok{ pten\_patho }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, residuals, var\_source,DMS\_score\_abundance)}
\NormalTok{pten\_somatic }\OtherTok{\textless{}{-}}\NormalTok{ pten\_somatic  }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, residuals, var\_source,DMS\_score\_abundance)}

\NormalTok{combined\_df }\OtherTok{\textless{}{-}} \FunctionTok{rbind}\NormalTok{(pten\_gnomad, pten\_patho, pten\_somatic)}
\FunctionTok{nrow}\NormalTok{(combined\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 189
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(combined\_df}\SpecialCharTok{$}\NormalTok{var\_source)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##     gnomAD Pathogenic    Somatic 
##         37        115         37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df\_clean\_2plot }\OtherTok{\textless{}{-}}\NormalTok{ combined\_df}
\NormalTok{median\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(var\_source) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{median\_residual =} \FunctionTok{median}\NormalTok{(residuals, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{),}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{()}
\NormalTok{  )}
\NormalTok{median\_df}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 3 x 3
##   var_source median_residual     n
##   <chr>                <dbl> <int>
## 1 Pathogenic        -1.48      115
## 2 Somatic           -1.52       37
## 3 gnomAD            -0.00905    37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df\_clean\_2plot}\SpecialCharTok{$}\NormalTok{var\_source }\OtherTok{\textless{}{-}} \FunctionTok{factor}\NormalTok{(}
\NormalTok{  pten\_df\_clean\_2plot}\SpecialCharTok{$}\NormalTok{var\_source,}
  \AttributeTok{levels =} \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Somatic"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{)}
\NormalTok{)}
\FunctionTok{table}\NormalTok{(pten\_df\_clean\_2plot}\SpecialCharTok{$}\NormalTok{var\_source)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##     gnomAD    Somatic Pathogenic 
##         37         37        115
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{custom\_colors }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \StringTok{"gnomAD"}     \OtherTok{=} \StringTok{"\#1b9e77"}\NormalTok{,  }\CommentTok{\# Teal green (unchanged)}
  \StringTok{"Somatic"}    \OtherTok{=} \StringTok{"\#fbb4ae"}\NormalTok{,  }\CommentTok{\# Soft sky blue}
  \StringTok{"Pathogenic"} \OtherTok{=} \StringTok{"\#de425b"}   \CommentTok{\# Deep ocean blue}
\NormalTok{)}


\NormalTok{p11 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(pten\_df\_clean\_2plot, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ residuals, }\AttributeTok{fill =}\NormalTok{ var\_source)) }\SpecialCharTok{+}
  \FunctionTok{geom\_violin}\NormalTok{(}\AttributeTok{trim =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{scale =} \StringTok{"width"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{color =} \ConstantTok{NA}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.15}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.7}\NormalTok{, }\AttributeTok{color =} \StringTok{"lightgrey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"crossbar"}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{fatten =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"point"}\NormalTok{, }\AttributeTok{shape =} \DecValTok{23}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{fill =} \StringTok{"black"}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ median\_residual }\SpecialCharTok{+} \DecValTok{1}\NormalTok{, }\AttributeTok{label =} \FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%.2f"}\NormalTok{, median\_residual)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{5}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =}\NormalTok{ custom\_colors) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =} \FloatTok{7.5}\NormalTok{, }\AttributeTok{label =} \FunctionTok{paste0}\NormalTok{(}\StringTok{"n = "}\NormalTok{, n)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \FloatTok{4.5}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"All 173 Variants"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{""}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Activity{-}abundance residuals"}\NormalTok{,}
    \AttributeTok{fill =} \StringTok{""}
\NormalTok{  ) }\SpecialCharTok{+} \FunctionTok{theme\_classic}\NormalTok{()}\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{legend.position =} \StringTok{"none"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{geom\_signif}\NormalTok{(}
    \AttributeTok{comparisons =} \FunctionTok{list}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Somatic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"Somatic"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{)),}
    \AttributeTok{map\_signif\_level =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{test =} \StringTok{"wilcox.test"}\NormalTok{,}
    \AttributeTok{step\_increase =} \FloatTok{0.1}\NormalTok{,}
    \AttributeTok{textsize =} \DecValTok{3}
\NormalTok{  ) }
\NormalTok{p11}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-10-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df\_clean\_2plot }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{mutation\_position =} \FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{str\_extract}\NormalTok{(mutant, }\StringTok{"(?\textless{}=}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{D)(}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{d+)(?=}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{D)"}\NormalTok{)))}

\NormalTok{pten\_df\_clean\_2plot\_int }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ all\_positions)}
\FunctionTok{nrow}\NormalTok{(pten\_df\_clean\_2plot\_int) }\CommentTok{\#89}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 89
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(pten\_df\_clean\_2plot\_int}\SpecialCharTok{$}\NormalTok{mutant)) }\CommentTok{\#89}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 89
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{median\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot\_int }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(var\_source) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{median\_residual =} \FunctionTok{median}\NormalTok{(residuals, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{),}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{()}
\NormalTok{  )}
\NormalTok{median\_df}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 3 x 3
##   var_source median_residual     n
##   <fct>                <dbl> <int>
## 1 gnomAD             -0.0190    14
## 2 Somatic            -2.38      15
## 3 Pathogenic         -1.77      60
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(pten\_df\_clean\_2plot\_int}\SpecialCharTok{$}\NormalTok{var\_source)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##     gnomAD    Somatic Pathogenic 
##         14         15         60
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{highlight\_muts }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"C124S"}\NormalTok{, }\StringTok{"G129E"}\NormalTok{, }\StringTok{"R130G"}\NormalTok{, }\StringTok{"R130Q"}\NormalTok{, }\StringTok{"P38S"}\NormalTok{, }\StringTok{"R130P"}\NormalTok{, }\StringTok{"D92H"}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_df\_clean\_2plot\_int)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 89
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p12 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(pten\_df\_clean\_2plot\_int, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ residuals, }\AttributeTok{fill =}\NormalTok{ var\_source)) }\SpecialCharTok{+}
  \FunctionTok{geom\_violin}\NormalTok{(}\AttributeTok{trim =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{scale =} \StringTok{"width"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{color =} \ConstantTok{NA}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.15}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.9}\NormalTok{, }\AttributeTok{color =} \StringTok{"lightgrey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"crossbar"}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{fatten =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"point"}\NormalTok{, }\AttributeTok{shape =} \DecValTok{23}\NormalTok{, }\AttributeTok{size =} \DecValTok{1}\NormalTok{, }\AttributeTok{fill =} \StringTok{"black"}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{geom\_point}\NormalTok{(}
    \AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df\_clean\_2plot\_int, mutant }\SpecialCharTok{\%in\%}\NormalTok{ highlight\_muts),}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ residuals),}
    \AttributeTok{shape =} \DecValTok{17}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{2}\NormalTok{,}
    \AttributeTok{fill =} \StringTok{"black"}\NormalTok{,}
    \AttributeTok{color =} \StringTok{"darkcyan"}\NormalTok{,}
    \AttributeTok{stroke =} \DecValTok{1}
\NormalTok{  ) }\SpecialCharTok{+} 
  \FunctionTok{geom\_text\_repel}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df\_clean\_2plot\_int, mutant }\SpecialCharTok{\%in\%}\NormalTok{ highlight\_muts),}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{label=}\NormalTok{mutant),}\AttributeTok{color =} \StringTok{"darkcyan"}\NormalTok{, }\AttributeTok{size =} \DecValTok{3}\NormalTok{)}\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ median\_residual }\SpecialCharTok{+} \DecValTok{1}\NormalTok{, }\AttributeTok{label =} \FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%.2f"}\NormalTok{, median\_residual)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{4}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =}\NormalTok{ custom\_colors) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =} \FloatTok{7.5}\NormalTok{, }\AttributeTok{label =} \FunctionTok{paste0}\NormalTok{(}\StringTok{"n = "}\NormalTok{, n)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \FloatTok{4.5}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"89 Orthosteric Variants"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{""}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Activity{-}abundance residuals"}\NormalTok{,}
    \AttributeTok{fill =} \StringTok{""}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{,}
    \AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle =} \DecValTok{45}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{)}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{geom\_signif}\NormalTok{(}
    \AttributeTok{comparisons =} \FunctionTok{list}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Somatic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"Somatic"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{)),}
    \AttributeTok{map\_signif\_level =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{test =} \StringTok{"wilcox.test"}\NormalTok{,}
    \AttributeTok{step\_increase =} \FloatTok{0.1}\NormalTok{,}
    \AttributeTok{textsize =} \DecValTok{3}
\NormalTok{  ) }\SpecialCharTok{+} \FunctionTok{ylim}\NormalTok{(}\SpecialCharTok{{-}}\DecValTok{8}\NormalTok{, }\DecValTok{11}\NormalTok{)}

\NormalTok{p12}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-11-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df\_clean\_2plot\_out }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ all\_positions)}
\FunctionTok{nrow}\NormalTok{(pten\_df\_clean\_2plot\_out) }\CommentTok{\#100}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 100
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{median\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot\_out }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(var\_source) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{median\_residual =} \FunctionTok{median}\NormalTok{(residuals, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{),}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{()}
\NormalTok{  )}
\NormalTok{median\_df}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 3 x 3
##   var_source median_residual     n
##   <fct>                <dbl> <int>
## 1 gnomAD              0.0154    23
## 2 Somatic            -1.33      22
## 3 Pathogenic         -1.01      55
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p13 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(pten\_df\_clean\_2plot\_out, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ residuals, }\AttributeTok{fill =}\NormalTok{ var\_source)) }\SpecialCharTok{+}
  \FunctionTok{geom\_violin}\NormalTok{(}\AttributeTok{trim =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{scale =} \StringTok{"width"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{color =} \ConstantTok{NA}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.15}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.9}\NormalTok{, }\AttributeTok{color =} \StringTok{"lightgrey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"crossbar"}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{fatten =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"point"}\NormalTok{, }\AttributeTok{shape =} \DecValTok{23}\NormalTok{, }\AttributeTok{size =} \DecValTok{1}\NormalTok{, }\AttributeTok{fill =} \StringTok{"black"}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{geom\_point}\NormalTok{(}
    \AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df\_clean\_2plot\_out, mutant }\SpecialCharTok{\%in\%}\NormalTok{ highlight\_muts),}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ residuals),}
    \AttributeTok{shape =} \DecValTok{17}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{2}\NormalTok{,}
    \AttributeTok{fill =} \StringTok{"black"}\NormalTok{,}
    \AttributeTok{color =} \StringTok{"darkcyan"}\NormalTok{,}
    \AttributeTok{stroke =} \DecValTok{1}
\NormalTok{  ) }\SpecialCharTok{+} 
  \FunctionTok{geom\_text\_repel}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(pten\_df\_clean\_2plot\_out, mutant }\SpecialCharTok{\%in\%}\NormalTok{ highlight\_muts),}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{label=}\NormalTok{mutant),}\AttributeTok{color =} \StringTok{"darkcyan"}\NormalTok{,}\AttributeTok{size =} \DecValTok{3}\NormalTok{)}\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ median\_residual }\SpecialCharTok{+} \DecValTok{1}\NormalTok{, }\AttributeTok{label =} \FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%.2f"}\NormalTok{, median\_residual)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{4}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =}\NormalTok{ custom\_colors) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =} \FloatTok{7.5}\NormalTok{, }\AttributeTok{label =} \FunctionTok{paste0}\NormalTok{(}\StringTok{"n = "}\NormalTok{, n)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \FloatTok{4.5}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"100 Non{-}orthosteric Variants"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{""}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Activity{-}abundance residuals"}\NormalTok{,}
    \AttributeTok{fill =} \StringTok{""}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{,}
    \AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle =} \DecValTok{45}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{)}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{geom\_signif}\NormalTok{(}
    \AttributeTok{comparisons =} \FunctionTok{list}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Somatic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"Somatic"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{)),}
    \AttributeTok{map\_signif\_level =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{test =} \StringTok{"wilcox.test"}\NormalTok{,}
    \AttributeTok{step\_increase =} \FloatTok{0.1}\NormalTok{,}
    \AttributeTok{textsize =} \DecValTok{3}
\NormalTok{  ) }\SpecialCharTok{+} \FunctionTok{ylim}\NormalTok{(}\SpecialCharTok{{-}}\DecValTok{8}\NormalTok{, }\DecValTok{11}\NormalTok{)}

\NormalTok{p13}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-12-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df\_clean\_2plot\_stable }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(DMS\_score\_abundance }\SpecialCharTok{\textgreater{}} \FloatTok{0.71}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(pten\_df\_clean\_2plot\_stable) }\CommentTok{\#81}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 81
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(pten\_df\_clean\_2plot\_stable}\SpecialCharTok{$}\NormalTok{var\_source)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##     gnomAD    Somatic Pathogenic 
##         28         14         39
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df\_clean\_2plot\_stable\_int }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot\_stable }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ all\_positions)}
\FunctionTok{nrow}\NormalTok{(pten\_df\_clean\_2plot\_stable\_int) }\CommentTok{\#48}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 48
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{median\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot\_stable\_int }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(var\_source) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{median\_residual =} \FunctionTok{median}\NormalTok{(residuals, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{),}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{()}
\NormalTok{  )}
\NormalTok{median\_df}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 3 x 3
##   var_source median_residual     n
##   <fct>                <dbl> <int>
## 1 gnomAD              -0.222     9
## 2 Somatic             -2.12      8
## 3 Pathogenic          -2.41     31
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p14 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(pten\_df\_clean\_2plot\_stable\_int, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ residuals, }\AttributeTok{fill =}\NormalTok{ var\_source)) }\SpecialCharTok{+}
  \FunctionTok{geom\_violin}\NormalTok{(}\AttributeTok{trim =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{scale =} \StringTok{"width"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{color =} \ConstantTok{NA}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.15}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.9}\NormalTok{, }\AttributeTok{color =} \StringTok{"lightgrey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"crossbar"}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{fatten =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"point"}\NormalTok{, }\AttributeTok{shape =} \DecValTok{23}\NormalTok{, }\AttributeTok{size =} \DecValTok{1}\NormalTok{, }\AttributeTok{fill =} \StringTok{"black"}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ median\_residual }\SpecialCharTok{+} \DecValTok{1}\NormalTok{, }\AttributeTok{label =} \FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%.2f"}\NormalTok{, median\_residual)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{5}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =}\NormalTok{ custom\_colors) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =} \FloatTok{7.5}\NormalTok{, }\AttributeTok{label =} \FunctionTok{paste0}\NormalTok{(}\StringTok{"n = "}\NormalTok{, n)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{4}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"48 WT{-}abundance Orthosteric Variants"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{""}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"LOESS fit residuals"}\NormalTok{,}
    \AttributeTok{fill =} \StringTok{""}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{,}
    \AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle =} \DecValTok{45}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{)}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{geom\_signif}\NormalTok{(}
    \AttributeTok{comparisons =} \FunctionTok{list}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Somatic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"Somatic"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{)),}
    \AttributeTok{map\_signif\_level =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{test =} \StringTok{"wilcox.test"}\NormalTok{,}
    \AttributeTok{step\_increase =} \FloatTok{0.1}\NormalTok{,}
    \AttributeTok{textsize =} \DecValTok{3}
\NormalTok{  ) }\SpecialCharTok{+} \FunctionTok{ylim}\NormalTok{(}\SpecialCharTok{{-}}\DecValTok{8}\NormalTok{, }\DecValTok{11}\NormalTok{)}

\NormalTok{p14}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Removed 12 rows containing missing values or values outside the scale range
## (`geom_violin()`).
\end{verbatim}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-13-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_df\_clean\_2plot\_stable\_out }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot\_stable }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ all\_positions)}
\FunctionTok{nrow}\NormalTok{(pten\_df\_clean\_2plot\_stable\_out) }\CommentTok{\#33}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 33
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{median\_df }\OtherTok{\textless{}{-}}\NormalTok{ pten\_df\_clean\_2plot\_stable\_out }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(var\_source) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{median\_residual =} \FunctionTok{median}\NormalTok{(residuals, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{),}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{()}
\NormalTok{  )}
\NormalTok{median\_df}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 3 x 3
##   var_source median_residual     n
##   <fct>                <dbl> <int>
## 1 gnomAD             -0.0127    19
## 2 Somatic            -1.34       6
## 3 Pathogenic         -1.56       8
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p15 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(pten\_df\_clean\_2plot\_stable\_out, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ residuals, }\AttributeTok{fill =}\NormalTok{ var\_source)) }\SpecialCharTok{+}
  \FunctionTok{geom\_violin}\NormalTok{(}\AttributeTok{trim =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{scale =} \StringTok{"width"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{color =} \ConstantTok{NA}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.15}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.9}\NormalTok{, }\AttributeTok{color =} \StringTok{"lightgrey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"crossbar"}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{fatten =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"point"}\NormalTok{, }\AttributeTok{shape =} \DecValTok{23}\NormalTok{, }\AttributeTok{size =} \DecValTok{1}\NormalTok{, }\AttributeTok{fill =} \StringTok{"black"}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =}\NormalTok{ median\_residual }\SpecialCharTok{+} \DecValTok{1}\NormalTok{, }\AttributeTok{label =} \FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%.2f"}\NormalTok{, median\_residual)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{4}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =}\NormalTok{ custom\_colors) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_source, }\AttributeTok{y =} \FloatTok{7.5}\NormalTok{, }\AttributeTok{label =} \FunctionTok{paste0}\NormalTok{(}\StringTok{"n = "}\NormalTok{, n)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \FloatTok{4.5}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"33 WT{-}abundance Non{-}orthosteric Variants"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{""}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"LOESS fit residuals"}\NormalTok{,}
    \AttributeTok{fill =} \StringTok{""}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{,}
    \AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle =} \DecValTok{45}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{)}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{geom\_signif}\NormalTok{(}
    \AttributeTok{comparisons =} \FunctionTok{list}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Somatic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"Somatic"}\NormalTok{, }\StringTok{"Pathogenic"}\NormalTok{)),}
    \AttributeTok{map\_signif\_level =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{test =} \StringTok{"wilcox.test"}\NormalTok{,}
    \AttributeTok{step\_increase =} \FloatTok{0.1}\NormalTok{,}
    \AttributeTok{textsize =} \DecValTok{3}
\NormalTok{  ) }\SpecialCharTok{+} \FunctionTok{ylim}\NormalTok{(}\SpecialCharTok{{-}}\DecValTok{8}\NormalTok{, }\DecValTok{11}\NormalTok{)}

\NormalTok{p15 }
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-13-2.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{range}\NormalTok{(pten\_df\_clean\_2plot}\SpecialCharTok{$}\NormalTok{residuals)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] -5.157748  2.422716
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{wilcox.test}\NormalTok{(pten\_df\_clean\_2plot\_stable\_int }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(var\_source }\SpecialCharTok{==} \StringTok{"gnomAD"}\NormalTok{)  }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{pull}\NormalTok{(residuals),}
\NormalTok{            pten\_df\_clean\_2plot\_stable\_int }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(var\_source }\SpecialCharTok{==} \StringTok{"Pathogenic"}\NormalTok{)  }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{pull}\NormalTok{(residuals))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Wilcoxon rank sum exact test
## 
## data:  pten_df_clean_2plot_stable_int %>% filter(var_source == "gnomAD") %>% pull(residuals) and pten_df_clean_2plot_stable_int %>% filter(var_source == "Pathogenic") %>% pull(residuals)
## W = 224, p-value = 0.004996
## alternative hypothesis: true location shift is not equal to 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p16 }\OtherTok{\textless{}{-}} \FunctionTok{plot\_grid}\NormalTok{(p12,p13,}\AttributeTok{ncol=}\DecValTok{2}\NormalTok{,}\AttributeTok{nrow=}\DecValTok{1}\NormalTok{)}
\NormalTok{p16}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-13-3.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/figs/panels/fig4\_violin3.pdf"}\NormalTok{, }
       \AttributeTok{plot =}\NormalTok{ p16, }\AttributeTok{width =} \DecValTok{5}\NormalTok{, }\AttributeTok{height =} \DecValTok{3}\NormalTok{, }\AttributeTok{dpi =} \DecValTok{300}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p17 }\OtherTok{\textless{}{-}} \FunctionTok{plot\_grid}\NormalTok{(p14,p15,}\AttributeTok{ncol=}\DecValTok{2}\NormalTok{,}\AttributeTok{nrow=}\DecValTok{1}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Removed 12 rows containing missing values or values outside the scale range
## (`geom_violin()`).
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p17}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-14-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/figs/panels/fig4\_violin4.pdf"}\NormalTok{, }
       \AttributeTok{plot =}\NormalTok{ p17, }\AttributeTok{width =} \DecValTok{6}\NormalTok{, }\AttributeTok{height =} \DecValTok{4}\NormalTok{, }\AttributeTok{dpi =} \DecValTok{300}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pdb }\OtherTok{\textless{}{-}} \FunctionTok{read.pdb}\NormalTok{(}\StringTok{"\textasciitilde{}/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/data/residual\_pdb/PTEN/1d5r.pdb"}\NormalTok{)}

\CommentTok{\# Extract C{-}alpha atoms from protein (exclude ligand)}
\NormalTok{protein\_ca }\OtherTok{\textless{}{-}}\NormalTok{ pdb}\SpecialCharTok{$}\NormalTok{atom[pdb}\SpecialCharTok{$}\NormalTok{atom}\SpecialCharTok{$}\NormalTok{elety }\SpecialCharTok{==} \StringTok{"CA"} \SpecialCharTok{\&}\NormalTok{ pdb}\SpecialCharTok{$}\NormalTok{atom}\SpecialCharTok{$}\NormalTok{resid }\SpecialCharTok{!=} \StringTok{"TLA"}\NormalTok{, ]}

\CommentTok{\# Extract ligand atoms (TLA residue)}
\NormalTok{ligand\_atoms }\OtherTok{\textless{}{-}}\NormalTok{ pdb}\SpecialCharTok{$}\NormalTok{atom[pdb}\SpecialCharTok{$}\NormalTok{atom}\SpecialCharTok{$}\NormalTok{resid }\SpecialCharTok{==} \StringTok{"TLA"} \SpecialCharTok{\&}\NormalTok{ pdb}\SpecialCharTok{$}\NormalTok{atom}\SpecialCharTok{$}\NormalTok{type }\SpecialCharTok{==} \StringTok{"HETATM"}\NormalTok{, ]}

\CommentTok{\# Calculate min distance from each C{-}alpha to the ligand}
\NormalTok{protein\_ca}\SpecialCharTok{$}\NormalTok{min\_dist\_to\_ligand }\OtherTok{\textless{}{-}} \FunctionTok{apply}\NormalTok{(protein\_ca, }\DecValTok{1}\NormalTok{, }\ControlFlowTok{function}\NormalTok{(atom) \{}
\NormalTok{  dists }\OtherTok{\textless{}{-}} \FunctionTok{sqrt}\NormalTok{((}\FunctionTok{as.numeric}\NormalTok{(atom[[}\StringTok{"x"}\NormalTok{]]) }\SpecialCharTok{{-}} \FunctionTok{as.numeric}\NormalTok{(ligand\_atoms}\SpecialCharTok{$}\NormalTok{x))}\SpecialCharTok{\^{}}\DecValTok{2} \SpecialCharTok{+}
\NormalTok{                  (}\FunctionTok{as.numeric}\NormalTok{(atom[[}\StringTok{"y"}\NormalTok{]]) }\SpecialCharTok{{-}} \FunctionTok{as.numeric}\NormalTok{(ligand\_atoms}\SpecialCharTok{$}\NormalTok{y))}\SpecialCharTok{\^{}}\DecValTok{2} \SpecialCharTok{+}
\NormalTok{                  (}\FunctionTok{as.numeric}\NormalTok{(atom[[}\StringTok{"z"}\NormalTok{]]) }\SpecialCharTok{{-}} \FunctionTok{as.numeric}\NormalTok{(ligand\_atoms}\SpecialCharTok{$}\NormalTok{z))}\SpecialCharTok{\^{}}\DecValTok{2}\NormalTok{)}
  \FunctionTok{return}\NormalTok{(}\FunctionTok{min}\NormalTok{(dists))}
\NormalTok{\})}

\CommentTok{\# View summary}
\FunctionTok{nrow}\NormalTok{(protein\_ca) }\CommentTok{\#307}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 307
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{summary}\NormalTok{(protein\_ca}\SpecialCharTok{$}\NormalTok{min\_dist\_to\_ligand)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   3.353  14.371  19.765  19.785  25.267  37.348
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(protein\_ca}\SpecialCharTok{$}\NormalTok{min\_dist\_to\_ligand)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 18.59907 17.19423 14.38261 15.13317 17.71047 20.54182
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(protein\_ca}\SpecialCharTok{$}\NormalTok{min\_dist\_to\_ligand) }\CommentTok{\#307}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 307
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{fil\_protein\_ca }\OtherTok{\textless{}{-}}\NormalTok{ protein\_ca }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(resid, resno, min\_dist\_to\_ligand)}
\NormalTok{merged\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_df\_clean, fil\_protein\_ca, }\AttributeTok{by.x=}\StringTok{"mutation\_position"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"resno"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(merged\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4839
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(merged\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   mutation_position mutant DMS_score_abundance DMS_score_bin_abundance
## 1                 2    T2A           0.7755378                       1
## 2                 2    T2D           1.1906667                       1
## 3                 2    T2K           0.9446276                       1
## 4                 2    T2N           1.1539191                       1
## 5                 2    T2P           0.8433099                       1
## 6                 2    T2R           0.9024487                       1
##       ESM.1v DMS_score_activity DMS_score_bin_activity V1      Model
## 1 -0.0355402        2.275491838                      1 20 ThermoMPNN
## 2 -9.5293701        2.387178001                      1 22 ThermoMPNN
## 3 -7.9349979       -0.009668751                      1 28 ThermoMPNN
## 4 -7.4877127       -3.204991850                      0 31 ThermoMPNN
## 5 -6.1493063       -0.257056982                      1 32 ThermoMPNN
## 6 -9.1065905        0.551584254                      1 34 ThermoMPNN
##                 Dataset    ddG_pred position wildtype mutation
## 1 AF-P60484-F1-model_v2  0.06013024        1        T        A
## 2 AF-P60484-F1-model_v2 -0.01170939        1        T        D
## 3 AF-P60484-F1-model_v2  0.19344133        1        T        K
## 4 AF-P60484-F1-model_v2  0.06487387        1        T        N
## 5 AF-P60484-F1-model_v2  0.16262442        1        T        P
## 6 AF-P60484-F1-model_v2  0.20207280        1        T        R
##                     pdb chain new_position uniprot exposure_SS exposure_rASA
## 1 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
## 2 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
## 3 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
## 4 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
## 5 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
## 6 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
##   spot_disorder func_esms_residue_class func_esms_variant_class
## 1             1                       1                       0
## 2             1                       1                       1
## 3             1                       1                       0
## 4             1                       1                       0
## 5             1                       1                       0
## 6             1                       1                       1
##   clinvar_clinical_significance
## 1                              
## 2                              
## 3                              
## 4                              
## 5                              
## 6                              
##                                                                                                                                                                                                                                                                                                                                                                                                              sequence
## 1 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 2 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 3 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 4 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 5 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
## 6 MTAIIKEIVSRNKRRYQEDGFDLDLTYIYPNIIAMGFPAERLEGVYRNNIDDVVRFLDSKHKNHYKIYNLCAERHYDTAKFNCRVAQYPFEDHNPPQLELIKPFCEDLDQWLSEDDNHVAAIHCKAGKGRTGVMICAYLLHRGKFLKAQEALDFYGEVRTRDKKGVTIPSQRRYVYYYSYLLKNHLDYRPVALLFHKMMFETIPMFSGGTCNPQFVVCQLKVKIYSSNSGPTRREDKFMYFEFPQPLPVCGDIKVEFFHKQNKMLKKDKMFHFWVNTFFIPGPEETSEKVENGSLCDQEIDSICSIERADNDKEYLVLTLTKNDLDKANKDKANRYFSPNFKVKLYFTKTVEEPSNPEASSSTSVTPDVSDNEPDHYRYSDTTDSDPENEPFDEDQHTQITKV
##   Phenotype_Class Phenotype_List Fitness_Score Abundance_Score HGVS.Consequence
## 1            <NA>           <NA>            NA              NA             <NA>
## 2            <NA>           <NA>            NA              NA             <NA>
## 3            <NA>           <NA>            NA              NA             <NA>
## 4            <NA>           <NA>            NA              NA             <NA>
## 5            <NA>           <NA>            NA              NA             <NA>
## 6            <NA>           <NA>            NA              NA             <NA>
##   ClinVar.Germline.Classification rsIDs ClinVar.Variation.ID gnomad     fitted
## 1                            <NA>  <NA>                   NA   <NA> -0.1669595
## 2                            <NA>  <NA>                   NA   <NA> -0.1979372
## 3                            <NA>  <NA>                   NA   <NA> -0.1787797
## 4                            <NA>  <NA>                   NA   <NA> -0.1899017
## 5                            <NA>  <NA>                   NA   <NA> -0.1742966
## 6                            <NA>  <NA>                   NA   <NA> -0.1759730
##     residuals somatic resid min_dist_to_ligand
## 1  2.44245136    <NA>  <NA>                 NA
## 2  2.58511520    <NA>  <NA>                 NA
## 3  0.16911091    <NA>  <NA>                 NA
## 4 -3.01509015    <NA>  <NA>                 NA
## 5 -0.08276039    <NA>  <NA>                 NA
## 6  0.72755725    <NA>  <NA>                 NA
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{sequence)}
\FunctionTok{nrow}\NormalTok{(merged\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4839
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(residuals }\SpecialCharTok{\textgreater{}=} \DecValTok{0}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(merged\_df) }\CommentTok{\#2449}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2390
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{all\_positions }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \DecValTok{88}\SpecialCharTok{:}\DecValTok{98}\NormalTok{,    }\CommentTok{\# WPD loop}
  \DecValTok{160}\SpecialCharTok{:}\DecValTok{171}\NormalTok{,  }\CommentTok{\# TI loop}
  \DecValTok{123}\SpecialCharTok{:}\DecValTok{130}\NormalTok{,  }\CommentTok{\# P loop}
  \DecValTok{35}\SpecialCharTok{:}\DecValTok{49}\NormalTok{,    }\CommentTok{\# Arginine loop}
  \DecValTok{200}\SpecialCharTok{:}\DecValTok{212}\NormalTok{,  }\CommentTok{\# CBR1}
  \DecValTok{226}\SpecialCharTok{:}\DecValTok{238}\NormalTok{,  }\CommentTok{\# CBR2}
  \DecValTok{258}\SpecialCharTok{:}\DecValTok{268}\NormalTok{,  }\CommentTok{\# CBR3}
  \DecValTok{327}\SpecialCharTok{:}\DecValTok{335}\NormalTok{,   }\CommentTok{\# CÎ±2 loop}
  \DecValTok{151}\SpecialCharTok{:}\DecValTok{174}\NormalTok{, }\CommentTok{\# membrane{-}binding alpha{-}helix}
  \DecValTok{1}\SpecialCharTok{:}\DecValTok{15} \CommentTok{\# PBM motif }
\NormalTok{)}

\NormalTok{active\_positions }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \DecValTok{88}\SpecialCharTok{:}\DecValTok{98}\NormalTok{,    }\CommentTok{\# WPD loop}
  \DecValTok{159}\SpecialCharTok{:}\DecValTok{171}\NormalTok{,  }\CommentTok{\# TI loop}
  \DecValTok{123}\SpecialCharTok{:}\DecValTok{130}\NormalTok{,  }\CommentTok{\# P loop}
  \DecValTok{35}\SpecialCharTok{:}\DecValTok{49}   \CommentTok{\# Arginine loop}
\NormalTok{)}

\NormalTok{binding\_positions }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \DecValTok{200}\SpecialCharTok{:}\DecValTok{212}\NormalTok{,  }\CommentTok{\# CBR1}
  \DecValTok{226}\SpecialCharTok{:}\DecValTok{238}\NormalTok{,  }\CommentTok{\# CBR2}
  \DecValTok{258}\SpecialCharTok{:}\DecValTok{268}\NormalTok{,  }\CommentTok{\# CBR3}
  \DecValTok{327}\SpecialCharTok{:}\DecValTok{335}\NormalTok{,   }\CommentTok{\# CÎ±2 loop}
  \DecValTok{151}\SpecialCharTok{:}\DecValTok{174}\NormalTok{, }\CommentTok{\# membrane{-}binding alpha{-}helix}
  \DecValTok{1}\SpecialCharTok{:}\DecValTok{15} \CommentTok{\# PBM motif }
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_df\_clean, fil\_protein\_ca, }\AttributeTok{by.x=}\StringTok{"mutation\_position"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"resno"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{merged\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{sequence)}
\FunctionTok{nrow}\NormalTok{(merged\_df) }\CommentTok{\#4839}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4839
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(merged\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   mutation_position mutant DMS_score_abundance DMS_score_bin_abundance
## 1                 2    T2A           0.7755378                       1
## 2                 2    T2D           1.1906667                       1
## 3                 2    T2K           0.9446276                       1
## 4                 2    T2N           1.1539191                       1
## 5                 2    T2P           0.8433099                       1
## 6                 2    T2R           0.9024487                       1
##       ESM.1v DMS_score_activity DMS_score_bin_activity V1      Model
## 1 -0.0355402        2.275491838                      1 20 ThermoMPNN
## 2 -9.5293701        2.387178001                      1 22 ThermoMPNN
## 3 -7.9349979       -0.009668751                      1 28 ThermoMPNN
## 4 -7.4877127       -3.204991850                      0 31 ThermoMPNN
## 5 -6.1493063       -0.257056982                      1 32 ThermoMPNN
## 6 -9.1065905        0.551584254                      1 34 ThermoMPNN
##                 Dataset    ddG_pred position wildtype mutation
## 1 AF-P60484-F1-model_v2  0.06013024        1        T        A
## 2 AF-P60484-F1-model_v2 -0.01170939        1        T        D
## 3 AF-P60484-F1-model_v2  0.19344133        1        T        K
## 4 AF-P60484-F1-model_v2  0.06487387        1        T        N
## 5 AF-P60484-F1-model_v2  0.16262442        1        T        P
## 6 AF-P60484-F1-model_v2  0.20207280        1        T        R
##                     pdb chain new_position uniprot exposure_SS exposure_rASA
## 1 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
## 2 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
## 3 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
## 4 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
## 5 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
## 6 AF-P60484-F1-model_v2     A            2  P60484           H          0.57
##   spot_disorder func_esms_residue_class func_esms_variant_class
## 1             1                       1                       0
## 2             1                       1                       1
## 3             1                       1                       0
## 4             1                       1                       0
## 5             1                       1                       0
## 6             1                       1                       1
##   clinvar_clinical_significance Phenotype_Class Phenotype_List Fitness_Score
## 1                                          <NA>           <NA>            NA
## 2                                          <NA>           <NA>            NA
## 3                                          <NA>           <NA>            NA
## 4                                          <NA>           <NA>            NA
## 5                                          <NA>           <NA>            NA
## 6                                          <NA>           <NA>            NA
##   Abundance_Score HGVS.Consequence ClinVar.Germline.Classification rsIDs
## 1              NA             <NA>                            <NA>  <NA>
## 2              NA             <NA>                            <NA>  <NA>
## 3              NA             <NA>                            <NA>  <NA>
## 4              NA             <NA>                            <NA>  <NA>
## 5              NA             <NA>                            <NA>  <NA>
## 6              NA             <NA>                            <NA>  <NA>
##   ClinVar.Variation.ID gnomad     fitted   residuals somatic resid
## 1                   NA   <NA> -0.1669595  2.44245136    <NA>  <NA>
## 2                   NA   <NA> -0.1979372  2.58511520    <NA>  <NA>
## 3                   NA   <NA> -0.1787797  0.16911091    <NA>  <NA>
## 4                   NA   <NA> -0.1899017 -3.01509015    <NA>  <NA>
## 5                   NA   <NA> -0.1742966 -0.08276039    <NA>  <NA>
## 6                   NA   <NA> -0.1759730  0.72755725    <NA>  <NA>
##   min_dist_to_ligand
## 1                 NA
## 2                 NA
## 3                 NA
## 4                 NA
## 5                 NA
## 6                 NA
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{sum}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(merged\_df}\SpecialCharTok{$}\NormalTok{min\_dist\_to\_ligand)) }\CommentTok{\#3738}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 3738
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(min\_dist\_to\_ligand))}

\NormalTok{patho\_df1 }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{ (mutant }\SpecialCharTok{\%in\%}\NormalTok{ patho\_df1}\SpecialCharTok{$}\NormalTok{mutant)}
\FunctionTok{nrow}\NormalTok{(patho\_df1) }\CommentTok{\#11}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 11
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{patho\_df2 }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{ (mutant }\SpecialCharTok{\%in\%}\NormalTok{ patho\_df2}\SpecialCharTok{$}\NormalTok{mutant)}
\FunctionTok{nrow}\NormalTok{(patho\_df2) }\CommentTok{\#11}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 10
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{patho\_df3 }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{ (mutant }\SpecialCharTok{\%in\%}\NormalTok{ patho\_df3}\SpecialCharTok{$}\NormalTok{mutant)}
\FunctionTok{nrow}\NormalTok{(patho\_df3) }\CommentTok{\#34}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 33
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_gnomad }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{ (mutant }\SpecialCharTok{\%in\%}\NormalTok{ pten\_gnomad}\SpecialCharTok{$}\NormalTok{mutant)}
\FunctionTok{nrow}\NormalTok{(pten\_gnomad) }\CommentTok{\#28}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 28
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pten\_somatic }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{ (mutant }\SpecialCharTok{\%in\%}\NormalTok{ pten\_somatic}\SpecialCharTok{$}\NormalTok{mutant)}
\FunctionTok{nrow}\NormalTok{(pten\_somatic) }\CommentTok{\#37}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{patho\_df1}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \StringTok{"ASD/DD"}
\NormalTok{patho\_df2}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \StringTok{"ASD/DD \& PHTS"}
\NormalTok{patho\_df3}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \StringTok{"PHTS"}
\NormalTok{pten\_gnomad}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \StringTok{"gnomAD"}
\NormalTok{pten\_somatic}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \StringTok{"somatic"}
\FunctionTok{colnames}\NormalTok{(pten\_gnomad)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "mutation_position"               "mutant"                         
##  [3] "DMS_score_abundance"             "DMS_score_bin_abundance"        
##  [5] "ESM.1v"                          "DMS_score_activity"             
##  [7] "DMS_score_bin_activity"          "V1"                             
##  [9] "Model"                           "Dataset"                        
## [11] "ddG_pred"                        "position"                       
## [13] "wildtype"                        "mutation"                       
## [15] "pdb"                             "chain"                          
## [17] "new_position"                    "uniprot"                        
## [19] "exposure_SS"                     "exposure_rASA"                  
## [21] "spot_disorder"                   "func_esms_residue_class"        
## [23] "func_esms_variant_class"         "clinvar_clinical_significance"  
## [25] "Phenotype_Class"                 "Phenotype_List"                 
## [27] "Fitness_Score"                   "Abundance_Score"                
## [29] "HGVS.Consequence"                "ClinVar.Germline.Classification"
## [31] "rsIDs"                           "ClinVar.Variation.ID"           
## [33] "gnomad"                          "fitted"                         
## [35] "residuals"                       "somatic"                        
## [37] "resid"                           "min_dist_to_ligand"             
## [39] "var_class2"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{patho\_df1 }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df1 }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, DMS\_score\_abundance, DMS\_score\_activity, var\_class2, residuals,min\_dist\_to\_ligand)}
\NormalTok{patho\_df2 }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df2 }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, DMS\_score\_abundance, DMS\_score\_activity, var\_class2,residuals,min\_dist\_to\_ligand)}
\NormalTok{patho\_df3 }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df3 }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, DMS\_score\_abundance, DMS\_score\_activity, var\_class2,residuals,min\_dist\_to\_ligand)}
\NormalTok{pten\_gnomad }\OtherTok{\textless{}{-}}\NormalTok{ pten\_gnomad }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, DMS\_score\_abundance, DMS\_score\_activity, var\_class2,residuals,min\_dist\_to\_ligand)}
\NormalTok{pten\_somatic }\OtherTok{\textless{}{-}}\NormalTok{ pten\_somatic }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(mutant, DMS\_score\_abundance, DMS\_score\_activity, var\_class2,residuals,min\_dist\_to\_ligand)}

\NormalTok{combined\_df }\OtherTok{\textless{}{-}} \FunctionTok{rbind}\NormalTok{(patho\_df1,patho\_df2,patho\_df3,pten\_gnomad, pten\_somatic)}
\FunctionTok{nrow}\NormalTok{(combined\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 119
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{median\_df }\OtherTok{\textless{}{-}}\NormalTok{ combined\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(var\_class2) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{median\_dist =} \FunctionTok{median}\NormalTok{(min\_dist\_to\_ligand, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{),}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{()}
\NormalTok{  )}
\NormalTok{median\_df}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 5 x 3
##   var_class2    median_dist     n
##   <chr>               <dbl> <int>
## 1 ASD/DD               17.6    11
## 2 ASD/DD & PHTS        13.6    10
## 3 PHTS                 14.7    33
## 4 gnomAD               20.4    28
## 5 somatic              15.0    37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(combined\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   mutant DMS_score_abundance DMS_score_activity var_class2  residuals
## 1   R14G         1.041911071         -0.7107802     ASD/DD -0.5318023
## 2   Q17E         0.971256096         -0.5467923     ASD/DD -0.3682351
## 3   H61Y         0.941901563         -0.5830217     ASD/DD -0.4042427
## 4   Y68C        -0.008473303         -2.7323465     ASD/DD  0.4783478
## 5   P95T         0.155054169         -0.4500599     ASD/DD  1.9428391
## 6  L139F        -0.126631454         -0.4326210     ASD/DD  3.3608216
##   min_dist_to_ligand
## 1           18.59907
## 2           15.13317
## 3           24.22288
## 4           12.59897
## 5            7.55193
## 6           17.61638
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{combined\_df}\SpecialCharTok{$}\NormalTok{var\_class2 }\OtherTok{\textless{}{-}} \FunctionTok{factor}\NormalTok{(}
\NormalTok{  combined\_df}\SpecialCharTok{$}\NormalTok{var\_class2,}
  \AttributeTok{levels =} \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"somatic"}\NormalTok{, }\StringTok{"ASD/DD"}\NormalTok{, }\StringTok{"ASD/DD \& PHTS"}\NormalTok{, }\StringTok{"PHTS"}\NormalTok{)}
\NormalTok{)}

\NormalTok{custom\_colors }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \StringTok{"gnomAD"}     \OtherTok{=} \StringTok{"\#1b9e77"}\NormalTok{,  }\CommentTok{\# Teal green (unchanged)}
  \StringTok{"somatic"}    \OtherTok{=} \StringTok{"\#fbb4ae"}\NormalTok{,  }\CommentTok{\# Soft sky blue}
  \StringTok{"ASD/DD"}   \OtherTok{=} \StringTok{"\#f4a6b3"}\NormalTok{,   }\CommentTok{\# Warm orange}
  \StringTok{"ASD/DD \& PHTS"}   \OtherTok{=} \StringTok{"\#e7298a"}\NormalTok{,   }\CommentTok{\# Muted purple}
  \StringTok{"PHTS"}   \OtherTok{=} \StringTok{"\#7570b3"}    \CommentTok{\# Hot pink / magenta}
\NormalTok{)}

\NormalTok{active\_positions }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \DecValTok{88}\SpecialCharTok{:}\DecValTok{98}\NormalTok{,    }\CommentTok{\# WPD loop}
  \DecValTok{159}\SpecialCharTok{:}\DecValTok{171}\NormalTok{,  }\CommentTok{\# TI loop}
  \DecValTok{35}\SpecialCharTok{:}\DecValTok{49}\NormalTok{,    }\CommentTok{\# Arginine loop}
  \DecValTok{123}\SpecialCharTok{:}\DecValTok{130}  \CommentTok{\# P loop}
\NormalTok{)}

\NormalTok{binding\_positions }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \DecValTok{200}\SpecialCharTok{:}\DecValTok{212}\NormalTok{,  }\CommentTok{\# CBR1}
  \DecValTok{226}\SpecialCharTok{:}\DecValTok{238}\NormalTok{,  }\CommentTok{\# CBR2}
  \DecValTok{258}\SpecialCharTok{:}\DecValTok{268}\NormalTok{,  }\CommentTok{\# CBR3}
  \DecValTok{327}\SpecialCharTok{:}\DecValTok{335}\NormalTok{,   }\CommentTok{\# CÎ±2 loop}
  \DecValTok{151}\SpecialCharTok{:}\DecValTok{174}\NormalTok{, }\CommentTok{\# membrane{-}binding alpha{-}helix}
  \DecValTok{1}\SpecialCharTok{:}\DecValTok{15} \CommentTok{\# PBM motif }
\NormalTok{)}

\NormalTok{combined\_df }\OtherTok{\textless{}{-}}\NormalTok{ combined\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{mutation\_position =} \FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{str\_extract}\NormalTok{(mutant, }\StringTok{"(?\textless{}=}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{D)(}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{d+)(?=}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{D)"}\NormalTok{)))}

\NormalTok{combined\_df}\SpecialCharTok{$}\NormalTok{site\_type }\OtherTok{\textless{}{-}} \StringTok{"Non{-}orthosteric site"}
\NormalTok{combined\_df}\SpecialCharTok{$}\NormalTok{site\_type[combined\_df}\SpecialCharTok{$}\NormalTok{mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ active\_positions] }\OtherTok{\textless{}{-}} \StringTok{"Active site"}
\NormalTok{combined\_df}\SpecialCharTok{$}\NormalTok{site\_type[combined\_df}\SpecialCharTok{$}\NormalTok{mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ binding\_positions] }\OtherTok{\textless{}{-}} \StringTok{"Binding site"}
\FunctionTok{table}\NormalTok{(combined\_df}\SpecialCharTok{$}\NormalTok{site\_type)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##          Active site         Binding site Non-orthosteric site 
##                   26                   22                   71
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p1 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(combined\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_class2, }\AttributeTok{y =}\NormalTok{ min\_dist\_to\_ligand, }\AttributeTok{fill =}\NormalTok{ var\_class2)) }\SpecialCharTok{+}
  \FunctionTok{geom\_violin}\NormalTok{(}\AttributeTok{trim =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{scale =} \StringTok{"width"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{color =} \ConstantTok{NA}\NormalTok{) }\SpecialCharTok{+}
 \FunctionTok{geom\_jitter}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{shape =}\NormalTok{ site\_type), }\AttributeTok{width =} \FloatTok{0.15}\NormalTok{, }\AttributeTok{size =} \FloatTok{2.5}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{,}
            \AttributeTok{fill =} \StringTok{"lightgrey"}\NormalTok{, }\AttributeTok{color =} \StringTok{"darkgrey"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.5}\NormalTok{)}\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"crossbar"}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{fatten =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"point"}\NormalTok{, }\AttributeTok{shape =} \DecValTok{23}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{fill =} \StringTok{"black"}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_class2, }\AttributeTok{y =}\NormalTok{  median\_dist }\SpecialCharTok{+} \DecValTok{1}\NormalTok{, }\AttributeTok{label =} \FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%.2f"}\NormalTok{,  median\_dist)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{5}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =}\NormalTok{ custom\_colors, }\AttributeTok{guide =} \StringTok{"none"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_shape\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}
  \StringTok{"Non{-}orthosteric site"} \OtherTok{=} \DecValTok{21}\NormalTok{,}
  \StringTok{"Active site"} \OtherTok{=} \DecValTok{22}\NormalTok{,}
  \StringTok{"Binding site"} \OtherTok{=} \DecValTok{23}
\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ median\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ var\_class2, }\AttributeTok{y =} \DecValTok{40}\NormalTok{, }\AttributeTok{label =} \FunctionTok{paste0}\NormalTok{(}\StringTok{"n = "}\NormalTok{, n)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \FloatTok{4.5}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"119 Variants"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{""}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Distance to TLA"}\NormalTok{,}
    \AttributeTok{fill =} \StringTok{""}\NormalTok{,}
    \AttributeTok{shape =} \StringTok{"Site Type"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{legend.position =} \StringTok{"right"}\NormalTok{,}
    \AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle =} \DecValTok{45}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{)}

\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{geom\_signif}\NormalTok{(}
    \AttributeTok{comparisons =} \FunctionTok{list}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"somatic"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"ASD/DD"}\NormalTok{), }
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"ASD/DD \& PHTS"}\NormalTok{),}
                       \FunctionTok{c}\NormalTok{(}\StringTok{"gnomAD"}\NormalTok{, }\StringTok{"PHTS"}\NormalTok{)),}
    \AttributeTok{map\_signif\_level =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{test =} \StringTok{"wilcox.test"}\NormalTok{,}
    \AttributeTok{step\_increase =} \FloatTok{0.1}\NormalTok{,}
    \AttributeTok{textsize =} \DecValTok{3}
\NormalTok{  ) }
\NormalTok{p1}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-16-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/figs/panels/fig4\_violin\_distance.pdf"}\NormalTok{, }
       \AttributeTok{plot =}\NormalTok{ p1, }\AttributeTok{width =} \DecValTok{6}\NormalTok{, }\AttributeTok{height =} \DecValTok{4}\NormalTok{, }\AttributeTok{dpi =} \DecValTok{300}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p1}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\begin{verbatim}
## Warning in wilcox.test.default(c(18.8445924869709, 18.8445924869709,
## 13.872890181934, : cannot compute exact p-value with ties
\end{verbatim}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-16-2.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_df\_clean, fil\_protein\_ca, }\AttributeTok{by.x=}\StringTok{"mutation\_position"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"resno"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{merged\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{sequence)}

\NormalTok{merged\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(min\_dist\_to\_ligand))}
\NormalTok{merged\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(residuals }\SpecialCharTok{\textless{}=} \DecValTok{0}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(merged\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2003
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df\_residue }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(mutation\_position) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{loess\_residual\_avg =} \FunctionTok{median}\NormalTok{(residuals, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{))}
    
\NormalTok{merged\_df\_residue }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(merged\_df\_residue, protein\_ca, }\AttributeTok{by.x=}\StringTok{"mutation\_position"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"resno"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
    
\NormalTok{merged\_df\_residue }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df\_residue[}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(merged\_df\_residue}\SpecialCharTok{$}\NormalTok{min\_dist\_to\_ligand), ]}
\FunctionTok{nrow}\NormalTok{(merged\_df\_residue) }\CommentTok{\#280}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 280
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Fit exponential decay: y = a * exp({-}b * x) + c}
\NormalTok{exp\_model }\OtherTok{\textless{}{-}} \FunctionTok{nls}\NormalTok{(}\FunctionTok{abs}\NormalTok{(loess\_residual\_avg) }\SpecialCharTok{\textasciitilde{}}\NormalTok{ a }\SpecialCharTok{*} \FunctionTok{exp}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{b }\SpecialCharTok{*}\NormalTok{ min\_dist\_to\_ligand),}
                 \AttributeTok{data =}\NormalTok{ merged\_df\_residue,}
                 \AttributeTok{start =} \FunctionTok{list}\NormalTok{(}\AttributeTok{a =} \DecValTok{1}\NormalTok{, }\AttributeTok{b =} \FloatTok{0.1}\NormalTok{))}
\NormalTok{exp\_model}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Nonlinear regression model
##   model: abs(loess_residual_avg) ~ a * exp(-b * min_dist_to_ligand)
##    data: merged_df_residue
##       a       b 
## 2.54954 0.04794 
##  residual sum-of-squares: 152.4
## 
## Number of iterations to convergence: 6 
## Achieved convergence tolerance: 3.505e-06
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Nonlinear regression model}
\CommentTok{\#   model: abs(loess\_residual\_avg) \textasciitilde{} a * exp({-}b * min\_dist\_to\_ligand)}
\CommentTok{\#    data: merged\_df\_residue}
\CommentTok{\#       a       b }
\CommentTok{\# 2.54954 0.04794 }
\CommentTok{\#  residual sum{-}of{-}squares: 152.4}
\CommentTok{\# }
\CommentTok{\# Number of iterations to convergence: 6 }
\CommentTok{\# Achieved convergence tolerance: 3.505e{-}06}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{a }\OtherTok{\textless{}{-}} \FloatTok{2.54954}
\NormalTok{b }\OtherTok{\textless{}{-}} \FloatTok{0.04794} 
\NormalTok{half\_d }\OtherTok{\textless{}{-}} \FunctionTok{log}\NormalTok{(}\DecValTok{2}\NormalTok{) }\SpecialCharTok{/}\NormalTok{ b  }
\NormalTok{half\_d }\CommentTok{\#14.45864}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 14.45864
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{y\_cutoff }\OtherTok{\textless{}{-}}\NormalTok{ a  }\SpecialCharTok{*} \FunctionTok{exp}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{ b }\SpecialCharTok{*}\NormalTok{ half\_d)}
\NormalTok{y\_cutoff}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1.27477
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{x\_vals }\OtherTok{\textless{}{-}} \FunctionTok{seq}\NormalTok{(}\FunctionTok{min}\NormalTok{(merged\_df\_residue}\SpecialCharTok{$}\NormalTok{min\_dist\_to\_ligand),}
              \FunctionTok{max}\NormalTok{(merged\_df\_residue}\SpecialCharTok{$}\NormalTok{min\_dist\_to\_ligand), }\AttributeTok{length.out =} \DecValTok{200}\NormalTok{)}

\CommentTok{\# {-}{-}{-} Bootstrapping for confidence intervals {-}{-}{-}}
\FunctionTok{set.seed}\NormalTok{(}\DecValTok{11}\NormalTok{)}
\NormalTok{boot\_params }\OtherTok{\textless{}{-}} \FunctionTok{replicate}\NormalTok{(}\DecValTok{1000}\NormalTok{, \{}
\NormalTok{  samp }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df\_residue[}\FunctionTok{sample}\NormalTok{(}\FunctionTok{nrow}\NormalTok{(merged\_df\_residue), }\AttributeTok{replace =} \ConstantTok{TRUE}\NormalTok{), ]}
\NormalTok{  fit }\OtherTok{\textless{}{-}} \FunctionTok{try}\NormalTok{(}\FunctionTok{nlsLM}\NormalTok{(}\FunctionTok{abs}\NormalTok{(loess\_residual\_avg) }\SpecialCharTok{\textasciitilde{}}\NormalTok{ a }\SpecialCharTok{*} \FunctionTok{exp}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{b }\SpecialCharTok{*}\NormalTok{ min\_dist\_to\_ligand),}
                   \AttributeTok{data =}\NormalTok{ samp, }\AttributeTok{start =} \FunctionTok{list}\NormalTok{(}\AttributeTok{a =} \DecValTok{1}\NormalTok{, }\AttributeTok{b =} \FloatTok{0.1}\NormalTok{)), }\AttributeTok{silent =} \ConstantTok{TRUE}\NormalTok{)}
  \ControlFlowTok{if}\NormalTok{ (}\FunctionTok{inherits}\NormalTok{(fit, }\StringTok{"try{-}error"}\NormalTok{)) }\FunctionTok{c}\NormalTok{(}\ConstantTok{NA}\NormalTok{, }\ConstantTok{NA}\NormalTok{) }\ControlFlowTok{else} \FunctionTok{coef}\NormalTok{(fit)}
\NormalTok{\})}
\NormalTok{boot\_params }\OtherTok{\textless{}{-}} \FunctionTok{t}\NormalTok{(boot\_params)[}\FunctionTok{complete.cases}\NormalTok{(}\FunctionTok{t}\NormalTok{(boot\_params)), ]}

\NormalTok{boot\_preds }\OtherTok{\textless{}{-}} \FunctionTok{apply}\NormalTok{(boot\_params, }\DecValTok{1}\NormalTok{, }\ControlFlowTok{function}\NormalTok{(p) p[}\DecValTok{1}\NormalTok{] }\SpecialCharTok{*} \FunctionTok{exp}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{p[}\DecValTok{2}\NormalTok{] }\SpecialCharTok{*}\NormalTok{ x\_vals))}

\NormalTok{fit\_df\_residue }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{min\_dist\_to\_ligand =}\NormalTok{ x\_vals,}
  \AttributeTok{loess\_residual\_avg =} \FunctionTok{predict}\NormalTok{(exp\_model, }\AttributeTok{newdata =} \FunctionTok{data.frame}\NormalTok{(}\AttributeTok{min\_dist\_to\_ligand =}\NormalTok{ x\_vals)),}
  \AttributeTok{lower =} \FunctionTok{apply}\NormalTok{(boot\_preds, }\DecValTok{1}\NormalTok{, quantile, }\AttributeTok{probs =} \FloatTok{0.025}\NormalTok{),}
  \AttributeTok{upper =} \FunctionTok{apply}\NormalTok{(boot\_preds, }\DecValTok{1}\NormalTok{, quantile, }\AttributeTok{probs =} \FloatTok{0.975}\NormalTok{)}
\NormalTok{)}


\FunctionTok{nrow}\NormalTok{(merged\_df\_residue) }\CommentTok{\#280}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 280
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df\_residue}\SpecialCharTok{$}\NormalTok{site\_type }\OtherTok{\textless{}{-}} \StringTok{"Non{-}orthosteric site"}
\NormalTok{merged\_df\_residue}\SpecialCharTok{$}\NormalTok{site\_type[}\FunctionTok{abs}\NormalTok{(merged\_df\_residue}\SpecialCharTok{$}\NormalTok{loess\_residual\_avg) }\SpecialCharTok{\textless{}=}\NormalTok{ y\_cutoff] }\OtherTok{\textless{}{-}} \StringTok{"Null"}
\NormalTok{merged\_df\_residue}\SpecialCharTok{$}\NormalTok{site\_type[merged\_df\_residue}\SpecialCharTok{$}\NormalTok{mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ binding\_positions] }\OtherTok{\textless{}{-}} \StringTok{"Binding site"}
\NormalTok{merged\_df\_residue}\SpecialCharTok{$}\NormalTok{site\_type[merged\_df\_residue}\SpecialCharTok{$}\NormalTok{mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ active\_positions] }\OtherTok{\textless{}{-}} \StringTok{"Active site"}

\FunctionTok{table}\NormalTok{(merged\_df\_residue}\SpecialCharTok{$}\NormalTok{site\_type)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##          Active site         Binding site Non-orthosteric site 
##                   45                   52                   62 
##                 Null 
##                  121
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{max}\NormalTok{(merged\_df\_residue }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{ (site\_type }\SpecialCharTok{==} \StringTok{"Active site"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{pull}\NormalTok{(min\_dist\_to\_ligand)) }\CommentTok{\#16.42677}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 16.42677
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Plot with fitted curve}
\NormalTok{p23 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(merged\_df\_residue, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ min\_dist\_to\_ligand, }\AttributeTok{y =} \FunctionTok{abs}\NormalTok{(loess\_residual\_avg))) }\SpecialCharTok{+}
  \CommentTok{\# First, plot NULL points separately with alpha = 0.2}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(merged\_df\_residue, site\_type }\SpecialCharTok{==} \StringTok{"Null"}\NormalTok{),}
             \FunctionTok{aes}\NormalTok{(}\AttributeTok{color =}\NormalTok{ site\_type), }\AttributeTok{size =} \DecValTok{1}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}
  \CommentTok{\# Then plot the rest}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(merged\_df\_residue, site\_type }\SpecialCharTok{!=} \StringTok{"Null"}\NormalTok{),}
             \FunctionTok{aes}\NormalTok{(}\AttributeTok{color =}\NormalTok{ site\_type), }\AttributeTok{size =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text\_repel}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(merged\_df\_residue, mutation\_position }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\DecValTok{123}\SpecialCharTok{:}\DecValTok{130}\NormalTok{)) ,}
                  \FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ mutation\_position, }\AttributeTok{color =}\NormalTok{ site\_type), }\AttributeTok{vjust =} \SpecialCharTok{{-}}\FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{3}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{geom\_text\_repel}\NormalTok{(}\AttributeTok{data =} \FunctionTok{subset}\NormalTok{(merged\_df\_residue, mutation\_position }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\DecValTok{14}\NormalTok{)) ,}
                  \FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ mutation\_position, }\AttributeTok{color =}\NormalTok{ site\_type), }\AttributeTok{vjust =} \SpecialCharTok{{-}}\FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{3}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text\_repel}\NormalTok{(}\AttributeTok{data =}\NormalTok{ merged\_df\_residue }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{ (loess\_residual\_avg }\SpecialCharTok{\textless{}=}\NormalTok{ y\_cutoff) }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{ (min\_dist\_to\_ligand }\SpecialCharTok{\textgreater{}} \FloatTok{16.42677}\NormalTok{),}
                  \FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ mutation\_position, }\AttributeTok{color =}\NormalTok{ site\_type), }\AttributeTok{vjust =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{3}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_ribbon}\NormalTok{(}\AttributeTok{data =}\NormalTok{ fit\_df\_residue,}
            \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ min\_dist\_to\_ligand, }\AttributeTok{ymin =}\NormalTok{ lower, }\AttributeTok{ymax =}\NormalTok{ upper),}
            \AttributeTok{fill =} \StringTok{"grey70"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.3}\NormalTok{, }\AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_line}\NormalTok{(}\AttributeTok{data =}\NormalTok{ fit\_df\_residue, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ min\_dist\_to\_ligand, }\AttributeTok{y =} \FunctionTok{abs}\NormalTok{(loess\_residual\_avg)),}
          \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{size =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_color\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}
    \StringTok{"Null"} \OtherTok{=} \StringTok{"grey"}\NormalTok{,}
    \StringTok{"Non{-}orthosteric site"} \OtherTok{=} \StringTok{"darkgreen"}\NormalTok{,}
    \StringTok{"Binding site"} \OtherTok{=} \StringTok{"cyan"}\NormalTok{,}
    \StringTok{"Active site"} \OtherTok{=} \StringTok{"orange"}
\NormalTok{  )) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{geom\_vline}\NormalTok{(}\AttributeTok{xintercept =} \FloatTok{16.42677}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{color =} \StringTok{"slategrey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_hline}\NormalTok{(}\AttributeTok{yintercept =}\NormalTok{ y\_cutoff, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{color =} \StringTok{"slategrey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"PTEN: per{-}residue allosteric decay"}\NormalTok{,}
    \AttributeTok{subtitle =} \StringTok{"280 residues"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{"Minimal distance to TLA"}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Median residual"}\NormalTok{,}
    \AttributeTok{color =} \StringTok{""}
\NormalTok{  )  }\SpecialCharTok{+} \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"bottom"}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{annotate}\NormalTok{(}\StringTok{"text"}\NormalTok{,  }\AttributeTok{x =} \ConstantTok{Inf}\NormalTok{, }\AttributeTok{y =} \ConstantTok{Inf}\NormalTok{,}
             \AttributeTok{hjust =} \DecValTok{1}\NormalTok{, }\AttributeTok{vjust =} \DecValTok{1}\NormalTok{,}
           \AttributeTok{label =} \StringTok{"y = a * exp (b * x)}\SpecialCharTok{\textbackslash{}n}\StringTok{a = 2.54954 }\SpecialCharTok{\textbackslash{}n}\StringTok{b = {-} 0.04794"}\NormalTok{,}
           \AttributeTok{size =} \DecValTok{4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{0}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
## i Please use `linewidth` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.
\end{verbatim}

\begin{verbatim}
## Warning: Duplicated aesthetics after name standardisation: hjust
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p23 }\OtherTok{\textless{}{-}} \FunctionTok{ggMarginal}\NormalTok{(}
\NormalTok{  p23,}
  \AttributeTok{type =} \StringTok{"density"}\NormalTok{,}
  \AttributeTok{margins =} \StringTok{"both"}\NormalTok{,}
  \AttributeTok{groupColour =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{groupFill =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{size =} \DecValTok{10}\NormalTok{,}
  \AttributeTok{colour =} \StringTok{"grey"}\NormalTok{,}
  \AttributeTok{fill =} \StringTok{"lightgrey"}
\NormalTok{)}
\FunctionTok{ggsave}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/figs/panels/fig4\_decay1.pdf"}\NormalTok{, }
       \AttributeTok{plot =}\NormalTok{ p23, }\AttributeTok{width =} \DecValTok{5}\NormalTok{, }\AttributeTok{height =} \DecValTok{5}\NormalTok{, }\AttributeTok{dpi =} \DecValTok{300}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: ggrepel: 143 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p23}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: ggrepel: 143 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps
\end{verbatim}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-18-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{lm\_model }\OtherTok{\textless{}{-}} \FunctionTok{lm}\NormalTok{(}\FunctionTok{log}\NormalTok{(}\FunctionTok{abs}\NormalTok{(loess\_residual\_avg)) }\SpecialCharTok{\textasciitilde{}}\NormalTok{ min\_dist\_to\_ligand, }\AttributeTok{data =}\NormalTok{ merged\_df\_residue)}
\FunctionTok{summary}\NormalTok{(lm\_model)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Call:
## lm(formula = log(abs(loess_residual_avg)) ~ min_dist_to_ligand, 
##     data = merged_df_residue)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -3.2854 -0.5895  0.0866  0.6599  1.5076 
## 
## Coefficients:
##                     Estimate Std. Error t value Pr(>|t|)    
## (Intercept)         0.405279   0.134745   3.008  0.00287 ** 
## min_dist_to_ligand -0.033466   0.006372  -5.252 2.99e-07 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 0.8119 on 278 degrees of freedom
## Multiple R-squared:  0.09027,    Adjusted R-squared:  0.087 
## F-statistic: 27.59 on 1 and 278 DF,  p-value: 2.992e-07
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Call:}
\CommentTok{\# lm(formula = log(abs(loess\_residual\_avg)) \textasciitilde{} min\_dist\_to\_ligand, }
\CommentTok{\#     data = merged\_df\_residue)}
\CommentTok{\# }
\CommentTok{\# Residuals:}
\CommentTok{\#     Min      1Q  Median      3Q     Max }
\CommentTok{\# {-}3.2854 {-}0.5895  0.0866  0.6599  1.5076 }
\CommentTok{\# }
\CommentTok{\# Coefficients:}
\CommentTok{\#                     Estimate Std. Error t value Pr(\textgreater{}|t|)    }
\CommentTok{\# (Intercept)         0.405279   0.134745   3.008  0.00287 ** }
\CommentTok{\# min\_dist\_to\_ligand {-}0.033466   0.006372  {-}5.252 2.99e{-}07 ***}
\CommentTok{\# {-}{-}{-}}
\CommentTok{\# Signif. codes:  0 â***â 0.001 â**â 0.01 â*â 0.05 â.â 0.1 â â 1}
\CommentTok{\# }
\CommentTok{\# Residual standard error: 0.8119 on 278 degrees of freedom}
\CommentTok{\# Multiple R{-}squared:  0.09027, Adjusted R{-}squared:  0.087 }
\CommentTok{\# F{-}statistic: 27.59 on 1 and 278 DF,  p{-}value: 2.992e{-}07}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(residuals }\SpecialCharTok{\textless{}=} \DecValTok{0}\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(merged\_df) }\CommentTok{\#2003}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2003
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Fit exponential decay: y = a * exp({-}b * x) + c}
\NormalTok{exp\_model }\OtherTok{\textless{}{-}} \FunctionTok{nls}\NormalTok{(}\FunctionTok{abs}\NormalTok{(residuals) }\SpecialCharTok{\textasciitilde{}}\NormalTok{ a }\SpecialCharTok{*} \FunctionTok{exp}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{b }\SpecialCharTok{*}\NormalTok{ min\_dist\_to\_ligand),}
                 \AttributeTok{data =}\NormalTok{ merged\_df,}
                 \AttributeTok{start =} \FunctionTok{list}\NormalTok{(}\AttributeTok{a =} \DecValTok{1}\NormalTok{, }\AttributeTok{b =} \FloatTok{0.1}\NormalTok{))}
\NormalTok{exp\_model}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Nonlinear regression model
##   model: abs(residuals) ~ a * exp(-b * min_dist_to_ligand)
##    data: merged_df
##       a       b 
## 2.99859 0.04776 
##  residual sum-of-squares: 2231
## 
## Number of iterations to convergence: 6 
## Achieved convergence tolerance: 9.093e-07
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Nonlinear regression model}
\CommentTok{\#   model: abs(residuals) \textasciitilde{} a * exp({-}b * min\_dist\_to\_ligand)}
\CommentTok{\#    data: merged\_df}
\CommentTok{\#       a       b }
\CommentTok{\# 2.99859 0.04776 }
\CommentTok{\#  residual sum{-}of{-}squares: 2231}
\CommentTok{\# }
\CommentTok{\# Number of iterations to convergence: 6 }
\CommentTok{\# Achieved convergence tolerance: 9.093e{-}07}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{a }\OtherTok{\textless{}{-}} \FloatTok{2.99859}
\NormalTok{b }\OtherTok{\textless{}{-}} \FloatTok{0.04776}  
\NormalTok{half\_d }\OtherTok{\textless{}{-}} \FunctionTok{log}\NormalTok{(}\DecValTok{2}\NormalTok{) }\SpecialCharTok{/}\NormalTok{ b  }
\NormalTok{half\_d }\CommentTok{\#14.51313}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 14.51313
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{y\_cutoff }\OtherTok{\textless{}{-}}\NormalTok{ a  }\SpecialCharTok{*} \FunctionTok{exp}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{b    }\SpecialCharTok{*}\NormalTok{ half\_d)}
\NormalTok{y\_cutoff}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1.499295
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df}\SpecialCharTok{$}\NormalTok{site\_type }\OtherTok{\textless{}{-}} \StringTok{"Non{-}orthosteric site"}
\NormalTok{merged\_df}\SpecialCharTok{$}\NormalTok{site\_type[}\FunctionTok{abs}\NormalTok{(merged\_df}\SpecialCharTok{$}\NormalTok{residuals) }\SpecialCharTok{\textless{}=}\NormalTok{ y\_cutoff] }\OtherTok{\textless{}{-}} \StringTok{"Null"}
\NormalTok{merged\_df}\SpecialCharTok{$}\NormalTok{site\_type[merged\_df}\SpecialCharTok{$}\NormalTok{mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ active\_positions] }\OtherTok{\textless{}{-}} \StringTok{"Active site"}
\NormalTok{merged\_df}\SpecialCharTok{$}\NormalTok{site\_type[merged\_df}\SpecialCharTok{$}\NormalTok{mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ binding\_positions] }\OtherTok{\textless{}{-}} \StringTok{"Binding site"}

\NormalTok{x\_vals }\OtherTok{\textless{}{-}} \FunctionTok{seq}\NormalTok{(}\FunctionTok{min}\NormalTok{(merged\_df}\SpecialCharTok{$}\NormalTok{min\_dist\_to\_ligand),}
              \FunctionTok{max}\NormalTok{(merged\_df}\SpecialCharTok{$}\NormalTok{min\_dist\_to\_ligand), }\AttributeTok{length.out =} \DecValTok{200}\NormalTok{)}

\CommentTok{\# {-}{-}{-} Bootstrapping for confidence intervals {-}{-}{-}}
\FunctionTok{set.seed}\NormalTok{(}\DecValTok{11}\NormalTok{)}
\NormalTok{boot\_params }\OtherTok{\textless{}{-}} \FunctionTok{replicate}\NormalTok{(}\DecValTok{1000}\NormalTok{, \{}
\NormalTok{  samp }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df[}\FunctionTok{sample}\NormalTok{(}\FunctionTok{nrow}\NormalTok{(merged\_df), }\AttributeTok{replace =} \ConstantTok{TRUE}\NormalTok{), ]}
\NormalTok{  fit }\OtherTok{\textless{}{-}} \FunctionTok{try}\NormalTok{(}\FunctionTok{nlsLM}\NormalTok{(}\FunctionTok{abs}\NormalTok{(residuals) }\SpecialCharTok{\textasciitilde{}}\NormalTok{ a }\SpecialCharTok{*} \FunctionTok{exp}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{b }\SpecialCharTok{*}\NormalTok{ min\_dist\_to\_ligand),}
                   \AttributeTok{data =}\NormalTok{ samp, }\AttributeTok{start =} \FunctionTok{list}\NormalTok{(}\AttributeTok{a =} \DecValTok{1}\NormalTok{, }\AttributeTok{b =} \FloatTok{0.1}\NormalTok{)), }\AttributeTok{silent =} \ConstantTok{TRUE}\NormalTok{)}
  \ControlFlowTok{if}\NormalTok{ (}\FunctionTok{inherits}\NormalTok{(fit, }\StringTok{"try{-}error"}\NormalTok{)) }\FunctionTok{c}\NormalTok{(}\ConstantTok{NA}\NormalTok{, }\ConstantTok{NA}\NormalTok{) }\ControlFlowTok{else} \FunctionTok{coef}\NormalTok{(fit)}
\NormalTok{\})}
\NormalTok{boot\_params }\OtherTok{\textless{}{-}} \FunctionTok{t}\NormalTok{(boot\_params)[}\FunctionTok{complete.cases}\NormalTok{(}\FunctionTok{t}\NormalTok{(boot\_params)), ]}

\NormalTok{boot\_preds }\OtherTok{\textless{}{-}} \FunctionTok{apply}\NormalTok{(boot\_params, }\DecValTok{1}\NormalTok{, }\ControlFlowTok{function}\NormalTok{(p) p[}\DecValTok{1}\NormalTok{] }\SpecialCharTok{*} \FunctionTok{exp}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{p[}\DecValTok{2}\NormalTok{] }\SpecialCharTok{*}\NormalTok{ x\_vals))}

\NormalTok{fit\_df\_residue }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{min\_dist\_to\_ligand =}\NormalTok{ x\_vals,}
  \AttributeTok{residuals =} \FunctionTok{predict}\NormalTok{(exp\_model, }\AttributeTok{newdata =} \FunctionTok{data.frame}\NormalTok{(}\AttributeTok{min\_dist\_to\_ligand =}\NormalTok{ x\_vals)),}
  \AttributeTok{lower =} \FunctionTok{apply}\NormalTok{(boot\_preds, }\DecValTok{1}\NormalTok{, quantile, }\AttributeTok{probs =} \FloatTok{0.025}\NormalTok{),}
  \AttributeTok{upper =} \FunctionTok{apply}\NormalTok{(boot\_preds, }\DecValTok{1}\NormalTok{, quantile, }\AttributeTok{probs =} \FloatTok{0.975}\NormalTok{)}
\NormalTok{)}

\NormalTok{merged\_df}\SpecialCharTok{$}\NormalTok{pathogenic\_status }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(}
\NormalTok{  merged\_df}\SpecialCharTok{$}\NormalTok{clinvar\_clinical\_significance }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"pathogenic"}\NormalTok{,}
                                           \StringTok{"likely\_pathogenic"}\NormalTok{),}
  \StringTok{"Pathogenic"}\NormalTok{, }\StringTok{"Other"}
\NormalTok{)}

\CommentTok{\# Plot}
\NormalTok{p21 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(merged\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ min\_dist\_to\_ligand, }\AttributeTok{y =} \FunctionTok{abs}\NormalTok{(residuals))) }\SpecialCharTok{+}
  \CommentTok{\# Base layer: all points}
  \FunctionTok{geom\_point}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{color =}\NormalTok{ site\_type, }\AttributeTok{alpha =}\NormalTok{ pathogenic\_status, }\AttributeTok{shape =}\NormalTok{ pathogenic\_status), }\AttributeTok{size =} \DecValTok{2}\NormalTok{) }\SpecialCharTok{+}
  
  \CommentTok{\# Loess fit line}
  \FunctionTok{geom\_ribbon}\NormalTok{(}\AttributeTok{data =}\NormalTok{ fit\_df\_residue,}
            \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ min\_dist\_to\_ligand, }\AttributeTok{ymin =}\NormalTok{ lower, }\AttributeTok{ymax =}\NormalTok{ upper),}
            \AttributeTok{fill =} \StringTok{"grey70"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.3}\NormalTok{, }\AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_line}\NormalTok{(}\AttributeTok{data =}\NormalTok{ fit\_df\_residue, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ min\_dist\_to\_ligand, }\AttributeTok{y =} \FunctionTok{abs}\NormalTok{(residuals)),}
          \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{size =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  
  \CommentTok{\# Repelled text labels for selected mutations}
  \FunctionTok{geom\_text\_repel}\NormalTok{(}
    \AttributeTok{data =} \FunctionTok{subset}\NormalTok{(merged\_df, pathogenic\_status }\SpecialCharTok{==} \StringTok{"Pathogenic"} \SpecialCharTok{\&}\NormalTok{ min\_dist\_to\_ligand }\SpecialCharTok{\textgreater{}} \FloatTok{16.42677} \SpecialCharTok{\&}\NormalTok{ residuals }\SpecialCharTok{\textless{}}\NormalTok{ y\_cutoff }\SpecialCharTok{\&}\NormalTok{ site\_type }\SpecialCharTok{==} \StringTok{"Non{-}orthosteric site"}\NormalTok{),}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ mutant,}\AttributeTok{color =}\NormalTok{ site\_type),}
    \AttributeTok{size =} \DecValTok{3}\NormalTok{, }
    \AttributeTok{max.overlaps =} \ConstantTok{Inf}\NormalTok{, }\AttributeTok{box.padding =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{point.padding =} \FloatTok{0.3}
\NormalTok{  ) }\SpecialCharTok{+}
  
  \CommentTok{\# Manual color palette for site type}
  \FunctionTok{scale\_color\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}
    \StringTok{"Null"} \OtherTok{=} \StringTok{"grey"}\NormalTok{,}
    \StringTok{"Non{-}orthosteric site"} \OtherTok{=} \StringTok{"darkgreen"}\NormalTok{,}
    \StringTok{"Binding site"} \OtherTok{=} \StringTok{"cyan"}\NormalTok{,}
    \StringTok{"Active site"} \OtherTok{=} \StringTok{"orange"}
\NormalTok{  )) }\SpecialCharTok{+}
  
  \CommentTok{\# Transparency scale}
  \FunctionTok{scale\_alpha\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"Other"} \OtherTok{=} \FloatTok{0.1}\NormalTok{, }\StringTok{"Pathogenic"} \OtherTok{=} \DecValTok{1}\NormalTok{)) }\SpecialCharTok{+}
  
  \CommentTok{\# Reference lines}
  \FunctionTok{geom\_vline}\NormalTok{(}\AttributeTok{xintercept =} \FloatTok{16.42677}\NormalTok{, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{color =} \StringTok{"slategrey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_hline}\NormalTok{(}\AttributeTok{yintercept =}\NormalTok{ y\_cutoff, }\AttributeTok{linetype =} \StringTok{"dashed"}\NormalTok{, }\AttributeTok{color =} \StringTok{"slategrey"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"Per{-}mutation Allosteric Decay"}\NormalTok{,}
    \AttributeTok{subtitle =} \StringTok{"2003 mutations with positive residuals"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{"Minimal distance to active sites (TLA)"}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"LOESS residual"}\NormalTok{,}
    \AttributeTok{color =} \StringTok{""}
\NormalTok{  ) }\SpecialCharTok{+} \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"bottom"}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{guides}\NormalTok{(}\AttributeTok{color =} \StringTok{"none"}\NormalTok{, }\AttributeTok{alpha =} \StringTok{"none"}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{annotate}\NormalTok{(}\StringTok{"text"}\NormalTok{,  }\AttributeTok{x =} \ConstantTok{Inf}\NormalTok{, }\AttributeTok{y =} \ConstantTok{Inf}\NormalTok{,}
             \AttributeTok{hjust =} \DecValTok{1}\NormalTok{, }\AttributeTok{vjust =} \DecValTok{1}\NormalTok{,}
           \AttributeTok{label =} \StringTok{"y = a * exp (b * x)}\SpecialCharTok{\textbackslash{}n}\StringTok{a = 2.99859 }\SpecialCharTok{\textbackslash{}n}\StringTok{b = {-} 0.04776"}\NormalTok{,}
           \AttributeTok{size =} \DecValTok{4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{0}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Duplicated aesthetics after name standardisation: hjust
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p21 }\OtherTok{\textless{}{-}} \FunctionTok{ggMarginal}\NormalTok{(}
\NormalTok{  p21,}
  \AttributeTok{type =} \StringTok{"density"}\NormalTok{,}
  \AttributeTok{margins =} \StringTok{"both"}\NormalTok{,}
  \AttributeTok{groupColour =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{groupFill =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{size =} \DecValTok{10}\NormalTok{,}
  \AttributeTok{colour =} \StringTok{"grey"}\NormalTok{,}
  \AttributeTok{fill =} \StringTok{"lightgrey"}
\NormalTok{)}

\FunctionTok{ggsave}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/figs/panels/fig4\_decay2.pdf"}\NormalTok{, }
       \AttributeTok{plot =}\NormalTok{ p21, }\AttributeTok{width =} \DecValTok{5}\NormalTok{, }\AttributeTok{height =} \DecValTok{5}\NormalTok{, }\AttributeTok{dpi =} \DecValTok{300}\NormalTok{)}
\NormalTok{p21}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-21-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{lm\_model }\OtherTok{\textless{}{-}} \FunctionTok{lm}\NormalTok{(}\FunctionTok{log}\NormalTok{(}\FunctionTok{abs}\NormalTok{(residuals)) }\SpecialCharTok{\textasciitilde{}}\NormalTok{ min\_dist\_to\_ligand, }\AttributeTok{data =}\NormalTok{ merged\_df)}
\FunctionTok{summary}\NormalTok{(lm\_model)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Call:
## lm(formula = log(abs(residuals)) ~ min_dist_to_ligand, data = merged_df)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -7.3797 -0.6332  0.3234  0.9096  2.2610 
## 
## Coefficients:
##                     Estimate Std. Error t value Pr(>|t|)    
## (Intercept)         0.662759   0.074530   8.892   <2e-16 ***
## min_dist_to_ligand -0.051687   0.003745 -13.804   <2e-16 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## Residual standard error: 1.268 on 2001 degrees of freedom
## Multiple R-squared:  0.08694,    Adjusted R-squared:  0.08649 
## F-statistic: 190.5 on 1 and 2001 DF,  p-value: < 2.2e-16
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Call:}
\CommentTok{\# lm(formula = log(abs(residuals)) \textasciitilde{} min\_dist\_to\_ligand, data = merged\_df)}
\CommentTok{\# }
\CommentTok{\# Residuals:}
\CommentTok{\#     Min      1Q  Median      3Q     Max }
\CommentTok{\# {-}7.3797 {-}0.6332  0.3234  0.9096  2.2610 }
\CommentTok{\# }
\CommentTok{\# Coefficients:}
\CommentTok{\#                     Estimate Std. Error t value Pr(\textgreater{}|t|)    }
\CommentTok{\# (Intercept)         0.662759   0.074530   8.892   \textless{}2e{-}16 ***}
\CommentTok{\# min\_dist\_to\_ligand {-}0.051687   0.003745 {-}13.804   \textless{}2e{-}16 ***}
\CommentTok{\# {-}{-}{-}}
\CommentTok{\# Signif. codes:  0 â***â 0.001 â**â 0.01 â*â 0.05 â.â 0.1 â â 1}
\CommentTok{\# }
\CommentTok{\# Residual standard error: 1.268 on 2001 degrees of freedom}
\CommentTok{\# Multiple R{-}squared:  0.08694, Adjusted R{-}squared:  0.08649 }
\CommentTok{\# F{-}statistic: 190.5 on 1 and 2001 DF,  p{-}value: \textless{} 2.2e{-}16}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(pten\_df\_clean, fil\_protein\_ca, }\AttributeTok{by.x=}\StringTok{"mutation\_position"}\NormalTok{, }\AttributeTok{by.y =} \StringTok{"resno"}\NormalTok{, }\AttributeTok{all.x =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{merged\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{sequence)}

\NormalTok{merged\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(min\_dist\_to\_ligand))}
\NormalTok{merged\_df}\SpecialCharTok{$}\NormalTok{pathogenic\_status }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(}
\NormalTok{  merged\_df}\SpecialCharTok{$}\NormalTok{clinvar\_clinical\_significance }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"pathogenic"}\NormalTok{,}
                                           \StringTok{"likely\_pathogenic"}\NormalTok{),}
  \StringTok{"Pathogenic"}\NormalTok{, }\StringTok{"Other"}
\NormalTok{)}
\FunctionTok{nrow}\NormalTok{(merged\_df)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 3738
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merged\_df\_int }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ all\_positions)}
\FunctionTok{nrow}\NormalTok{(merged\_df\_int) }\CommentTok{\#1279}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1279
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(merged\_df\_int}\SpecialCharTok{$}\NormalTok{pathogenic\_status)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##      Other Pathogenic 
##       1224         55
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
 \CommentTok{\# Other Pathogenic }
 \CommentTok{\#      1224         55 }

\NormalTok{merged\_df\_out }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{mutation\_position }\SpecialCharTok{\%in\%}\NormalTok{ all\_positions)}
\FunctionTok{nrow}\NormalTok{(merged\_df\_out) }\CommentTok{\#2459}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2459
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(merged\_df\_out}\SpecialCharTok{$}\NormalTok{pathogenic\_status)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##      Other Pathogenic 
##       2404         55
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
     \CommentTok{\# Other Pathogenic }
     \CommentTok{\#  2404         55 }
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# 1. Set up}
\FunctionTok{set.seed}\NormalTok{(}\DecValTok{123}\NormalTok{)}
\NormalTok{n\_boot }\OtherTok{\textless{}{-}} \DecValTok{1000}
\NormalTok{match\_window }\OtherTok{\textless{}{-}} \FloatTok{0.05}
\NormalTok{n\_patho }\OtherTok{\textless{}{-}} \FunctionTok{nrow}\NormalTok{(merged\_df\_int }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(pathogenic\_status }\SpecialCharTok{==} \StringTok{"Pathogenic"}\NormalTok{))}

\CommentTok{\# 2. Get abundance/residuals from pten\_patho}
\NormalTok{patho\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df\_int }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(pathogenic\_status }\SpecialCharTok{==} \StringTok{"Pathogenic"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(DMS\_score\_abundance, residuals)}
\NormalTok{patho\_df}\SpecialCharTok{$}\NormalTok{group }\OtherTok{\textless{}{-}} \StringTok{"Pathogenic"}

\CommentTok{\# 3. Bootstrap sampling from non{-}patho pool with abundance matching}
\NormalTok{non\_patho\_pool }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df\_int }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(pathogenic\_status }\SpecialCharTok{!=} \StringTok{"Pathogenic"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{(mutant }\SpecialCharTok{\%in\%}\NormalTok{ pten\_patho}\SpecialCharTok{$}\NormalTok{mutant)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{(mutant }\SpecialCharTok{\%in\%}\NormalTok{ pten\_somatic}\SpecialCharTok{$}\NormalTok{mutant)) }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(DMS\_score\_abundance, residuals)}

\NormalTok{bootstrap\_medians }\OtherTok{\textless{}{-}} \FunctionTok{vector}\NormalTok{(}\StringTok{"numeric"}\NormalTok{, }\AttributeTok{length =}\NormalTok{ n\_boot)}

\CommentTok{\# Pre{-}group non{-}patho pool into bins}
\NormalTok{non\_patho\_pool }\OtherTok{\textless{}{-}}\NormalTok{ non\_patho\_pool }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{bin =} \FunctionTok{cut}\NormalTok{(DMS\_score\_abundance, }\AttributeTok{breaks =} \FunctionTok{seq}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{0.5}\NormalTok{, }\DecValTok{2}\NormalTok{, }\AttributeTok{by =}\NormalTok{ match\_window)))}

\CommentTok{\# Bin pathogenic variants accordingly}
\NormalTok{patho\_df }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{bin =} \FunctionTok{cut}\NormalTok{(DMS\_score\_abundance, }\AttributeTok{breaks =} \FunctionTok{seq}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{0.5}\NormalTok{, }\DecValTok{2}\NormalTok{, }\AttributeTok{by =}\NormalTok{ match\_window)))}

\CommentTok{\# Create lookup table for fast sampling}
\NormalTok{bin\_lookup }\OtherTok{\textless{}{-}} \FunctionTok{split}\NormalTok{(non\_patho\_pool}\SpecialCharTok{$}\NormalTok{residuals, non\_patho\_pool}\SpecialCharTok{$}\NormalTok{bin)}

\CommentTok{\# Bootstrap matrix}
\NormalTok{bootstrap\_matrix }\OtherTok{\textless{}{-}} \FunctionTok{matrix}\NormalTok{(}\ConstantTok{NA}\NormalTok{, }\AttributeTok{nrow =}\NormalTok{ n\_boot, }\AttributeTok{ncol =}\NormalTok{ n\_patho)}

\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\NormalTok{n\_boot) \{}
  \ControlFlowTok{for}\NormalTok{ (j }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\NormalTok{n\_patho) \{}
\NormalTok{    bin\_j }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df}\SpecialCharTok{$}\NormalTok{bin[j]}
\NormalTok{    candidates }\OtherTok{\textless{}{-}}\NormalTok{ bin\_lookup[[}\FunctionTok{as.character}\NormalTok{(bin\_j)]]}
    \ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{is.null}\NormalTok{(candidates) }\SpecialCharTok{\&\&} \FunctionTok{length}\NormalTok{(candidates) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) \{}
\NormalTok{      bootstrap\_matrix[i, j] }\OtherTok{\textless{}{-}} \FunctionTok{sample}\NormalTok{(candidates, }\DecValTok{1}\NormalTok{)}
\NormalTok{    \}}
\NormalTok{  \}}
\NormalTok{\}}

\CommentTok{\# Summarize into a dataframe}
\NormalTok{boot\_df }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{group =} \StringTok{"Random abundance{-}matched"}\NormalTok{,}
  \AttributeTok{residuals =} \FunctionTok{apply}\NormalTok{(bootstrap\_matrix, }\DecValTok{1}\NormalTok{, median, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)   }\CommentTok{\# Mean across matches per bootstrap}
\NormalTok{)}

\CommentTok{\# Combine with patho residuals}
\NormalTok{plot\_df }\OtherTok{\textless{}{-}} \FunctionTok{bind\_rows}\NormalTok{(}
\NormalTok{  patho\_df }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(group, residuals),}
\NormalTok{  boot\_df}
\NormalTok{)}

\NormalTok{label\_df }\OtherTok{\textless{}{-}}\NormalTok{ plot\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(group) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{(),}
    \AttributeTok{median\_val =} \FunctionTok{median}\NormalTok{(residuals),}
    \AttributeTok{y\_max =} \FunctionTok{max}\NormalTok{(residuals),}
    \AttributeTok{.groups =} \StringTok{"drop"}
\NormalTok{  )}

\NormalTok{label\_df }\OtherTok{\textless{}{-}}\NormalTok{ label\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{n\_label =} \FunctionTok{case\_when}\NormalTok{(}
\NormalTok{    group }\SpecialCharTok{==} \StringTok{"Random abundance{-}matched"} \SpecialCharTok{\textasciitilde{}} \StringTok{"bootstrapped 1000 times"}\NormalTok{,}
    \ConstantTok{TRUE} \SpecialCharTok{\textasciitilde{}} \FunctionTok{paste0}\NormalTok{(}\StringTok{"n = "}\NormalTok{, n)}
\NormalTok{  ))}

\CommentTok{\# Plot}
\NormalTok{p\_fast }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(plot\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ group, }\AttributeTok{y =}\NormalTok{ residuals, }\AttributeTok{fill =}\NormalTok{ group)) }\SpecialCharTok{+}
  \FunctionTok{geom\_violin}\NormalTok{(}\AttributeTok{trim =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{scale =} \StringTok{"width"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{color =} \ConstantTok{NA}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.15}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.7}\NormalTok{, }\AttributeTok{color =} \StringTok{"lightgrey"}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"crossbar"}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{fatten =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"point"}\NormalTok{, }\AttributeTok{shape =} \DecValTok{23}\NormalTok{, }\AttributeTok{size =} \FloatTok{2.5}\NormalTok{,}
               \AttributeTok{fill =} \StringTok{"black"}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}

 \FunctionTok{geom\_text}\NormalTok{(}
  \AttributeTok{data =}\NormalTok{ label\_df,}
  \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ group, }\AttributeTok{y =} \FloatTok{4.5}\NormalTok{, }\AttributeTok{label =}\NormalTok{ n\_label),}
  \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{size =} \DecValTok{4}\NormalTok{) }\SpecialCharTok{+}

  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ label\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ group, }\AttributeTok{y =}\NormalTok{ median\_val }\SpecialCharTok{+} \FloatTok{0.25}\NormalTok{, }\AttributeTok{label =} \FunctionTok{sprintf}\NormalTok{(}\StringTok{" \%.2f"}\NormalTok{, median\_val)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{6}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{x =} \StringTok{""}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Activity{-}abundance residuals"}\NormalTok{,}
    \AttributeTok{title =} \StringTok{"Orthosteric variants"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{(}\AttributeTok{base\_size =} \DecValTok{14}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"Pathogenic"} \OtherTok{=} \StringTok{"cornflowerblue"}\NormalTok{, }\StringTok{"Random abundance{-}matched"} \OtherTok{=} \StringTok{"darkolivegreen3"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_signif}\NormalTok{(}\AttributeTok{comparisons =} \FunctionTok{list}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"Pathogenic"}\NormalTok{, }\StringTok{"Random abundance{-}matched"}\NormalTok{)),}
              \AttributeTok{map\_signif\_level =} \ConstantTok{FALSE}\NormalTok{,}
              \AttributeTok{test =} \StringTok{"wilcox.test"}\NormalTok{,}
              \AttributeTok{tip\_length =} \FloatTok{0.01}\NormalTok{)}


\NormalTok{p\_fast}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-24-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# 1. Set up}
\FunctionTok{set.seed}\NormalTok{(}\DecValTok{123}\NormalTok{)}
\NormalTok{n\_boot }\OtherTok{\textless{}{-}} \DecValTok{1000}
\NormalTok{match\_window }\OtherTok{\textless{}{-}} \FloatTok{0.05}
\NormalTok{n\_patho }\OtherTok{\textless{}{-}} \FunctionTok{nrow}\NormalTok{(merged\_df\_out }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(pathogenic\_status }\SpecialCharTok{==} \StringTok{"Pathogenic"}\NormalTok{))}

\CommentTok{\# 2. Get abundance/residuals from pten\_patho}
\NormalTok{patho\_df }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df\_out }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(pathogenic\_status }\SpecialCharTok{==} \StringTok{"Pathogenic"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(DMS\_score\_abundance, residuals)}
\NormalTok{patho\_df}\SpecialCharTok{$}\NormalTok{group }\OtherTok{\textless{}{-}} \StringTok{"Pathogenic"}

\CommentTok{\# 3. Bootstrap sampling from non{-}patho pool with abundance matching}
\NormalTok{non\_patho\_pool }\OtherTok{\textless{}{-}}\NormalTok{ merged\_df\_out }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(pathogenic\_status }\SpecialCharTok{!=} \StringTok{"Pathogenic"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{(mutant }\SpecialCharTok{\%in\%}\NormalTok{ pten\_patho}\SpecialCharTok{$}\NormalTok{mutant)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\NormalTok{(mutant }\SpecialCharTok{\%in\%}\NormalTok{ pten\_somatic}\SpecialCharTok{$}\NormalTok{mutant)) }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(DMS\_score\_abundance, residuals)}

\NormalTok{bootstrap\_medians }\OtherTok{\textless{}{-}} \FunctionTok{vector}\NormalTok{(}\StringTok{"numeric"}\NormalTok{, }\AttributeTok{length =}\NormalTok{ n\_boot)}

\CommentTok{\# Pre{-}group non{-}patho pool into bins}
\NormalTok{non\_patho\_pool }\OtherTok{\textless{}{-}}\NormalTok{ non\_patho\_pool }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{bin =} \FunctionTok{cut}\NormalTok{(DMS\_score\_abundance, }\AttributeTok{breaks =} \FunctionTok{seq}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{0.5}\NormalTok{, }\DecValTok{2}\NormalTok{, }\AttributeTok{by =}\NormalTok{ match\_window)))}

\CommentTok{\# Bin pathogenic variants accordingly}
\NormalTok{patho\_df }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{bin =} \FunctionTok{cut}\NormalTok{(DMS\_score\_abundance, }\AttributeTok{breaks =} \FunctionTok{seq}\NormalTok{(}\SpecialCharTok{{-}}\FloatTok{0.5}\NormalTok{, }\DecValTok{2}\NormalTok{, }\AttributeTok{by =}\NormalTok{ match\_window)))}

\CommentTok{\# Create lookup table for fast sampling}
\NormalTok{bin\_lookup }\OtherTok{\textless{}{-}} \FunctionTok{split}\NormalTok{(non\_patho\_pool}\SpecialCharTok{$}\NormalTok{residuals, non\_patho\_pool}\SpecialCharTok{$}\NormalTok{bin)}

\CommentTok{\# Bootstrap matrix}
\NormalTok{bootstrap\_matrix }\OtherTok{\textless{}{-}} \FunctionTok{matrix}\NormalTok{(}\ConstantTok{NA}\NormalTok{, }\AttributeTok{nrow =}\NormalTok{ n\_boot, }\AttributeTok{ncol =}\NormalTok{ n\_patho)}

\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\NormalTok{n\_boot) \{}
  \ControlFlowTok{for}\NormalTok{ (j }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\NormalTok{n\_patho) \{}
\NormalTok{    bin\_j }\OtherTok{\textless{}{-}}\NormalTok{ patho\_df}\SpecialCharTok{$}\NormalTok{bin[j]}
\NormalTok{    candidates }\OtherTok{\textless{}{-}}\NormalTok{ bin\_lookup[[}\FunctionTok{as.character}\NormalTok{(bin\_j)]]}
    \ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{is.null}\NormalTok{(candidates) }\SpecialCharTok{\&\&} \FunctionTok{length}\NormalTok{(candidates) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) \{}
\NormalTok{      bootstrap\_matrix[i, j] }\OtherTok{\textless{}{-}} \FunctionTok{sample}\NormalTok{(candidates, }\DecValTok{1}\NormalTok{)}
\NormalTok{    \}}
\NormalTok{  \}}
\NormalTok{\}}

\CommentTok{\# Summarize into a dataframe}
\NormalTok{boot\_df }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{group =} \StringTok{"Random abundance{-}matched"}\NormalTok{,}
  \AttributeTok{residuals =} \FunctionTok{apply}\NormalTok{(bootstrap\_matrix, }\DecValTok{1}\NormalTok{, median, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)   }\CommentTok{\# Mean across matches per bootstrap}
\NormalTok{)}

\CommentTok{\# Combine with patho residuals}
\NormalTok{plot\_df }\OtherTok{\textless{}{-}} \FunctionTok{bind\_rows}\NormalTok{(}
\NormalTok{  patho\_df }\SpecialCharTok{\%\textgreater{}\%}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(group, residuals),}
\NormalTok{  boot\_df}
\NormalTok{)}

\NormalTok{label\_df }\OtherTok{\textless{}{-}}\NormalTok{ plot\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{group\_by}\NormalTok{(group) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{summarise}\NormalTok{(}
    \AttributeTok{n =} \FunctionTok{n}\NormalTok{(),}
    \AttributeTok{median\_val =} \FunctionTok{median}\NormalTok{(residuals),}
    \AttributeTok{y\_max =} \FunctionTok{max}\NormalTok{(residuals),}
    \AttributeTok{.groups =} \StringTok{"drop"}
\NormalTok{  )}

\NormalTok{label\_df }\OtherTok{\textless{}{-}}\NormalTok{ label\_df }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{n\_label =} \FunctionTok{case\_when}\NormalTok{(}
\NormalTok{    group }\SpecialCharTok{==} \StringTok{"Random abundance{-}matched"} \SpecialCharTok{\textasciitilde{}} \StringTok{"bootstrapped 1000 times"}\NormalTok{,}
    \ConstantTok{TRUE} \SpecialCharTok{\textasciitilde{}} \FunctionTok{paste0}\NormalTok{(}\StringTok{"n = "}\NormalTok{, n)}
\NormalTok{  ))}

\CommentTok{\# Plot}
\NormalTok{p\_fast2 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(plot\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ group, }\AttributeTok{y =}\NormalTok{ residuals, }\AttributeTok{fill =}\NormalTok{ group)) }\SpecialCharTok{+}
  \FunctionTok{geom\_violin}\NormalTok{(}\AttributeTok{trim =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{scale =} \StringTok{"width"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{color =} \ConstantTok{NA}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_jitter}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.15}\NormalTok{, }\AttributeTok{size =} \DecValTok{2}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.7}\NormalTok{, }\AttributeTok{color =} \StringTok{"lightgrey"}\NormalTok{) }\SpecialCharTok{+}
    \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"crossbar"}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.4}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{fatten =} \DecValTok{1}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{stat\_summary}\NormalTok{(}\AttributeTok{fun =}\NormalTok{ median, }\AttributeTok{geom =} \StringTok{"point"}\NormalTok{, }\AttributeTok{shape =} \DecValTok{23}\NormalTok{, }\AttributeTok{size =} \FloatTok{2.5}\NormalTok{,}
               \AttributeTok{fill =} \StringTok{"black"}\NormalTok{, }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{stroke =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}

 \FunctionTok{geom\_text}\NormalTok{(}
  \AttributeTok{data =}\NormalTok{ label\_df,}
  \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ group, }\AttributeTok{y =} \FloatTok{4.5}\NormalTok{, }\AttributeTok{label =}\NormalTok{ n\_label),}
  \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{size =} \DecValTok{4}\NormalTok{) }\SpecialCharTok{+}

  \FunctionTok{geom\_text}\NormalTok{(}
    \AttributeTok{data =}\NormalTok{ label\_df,}
    \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ group, }\AttributeTok{y =}\NormalTok{ median\_val }\SpecialCharTok{+} \FloatTok{0.25}\NormalTok{, }\AttributeTok{label =} \FunctionTok{sprintf}\NormalTok{(}\StringTok{" \%.2f"}\NormalTok{, median\_val)),}
    \AttributeTok{inherit.aes =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{size =} \DecValTok{6}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{x =} \StringTok{""}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"Activity{-}abundance residual"}\NormalTok{,}
    \AttributeTok{title =} \StringTok{"Non{-}orthosteric variants"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{theme\_classic}\NormalTok{(}\AttributeTok{base\_size =} \DecValTok{14}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"Pathogenic"} \OtherTok{=} \StringTok{"cornflowerblue"}\NormalTok{, }\StringTok{"Random abundance{-}matched"} \OtherTok{=} \StringTok{"darkolivegreen3"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_signif}\NormalTok{(}\AttributeTok{comparisons =} \FunctionTok{list}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"Pathogenic"}\NormalTok{, }\StringTok{"Random abundance{-}matched"}\NormalTok{)),}
              \AttributeTok{map\_signif\_level =} \ConstantTok{FALSE}\NormalTok{,}
              \AttributeTok{test =} \StringTok{"wilcox.test"}\NormalTok{,}
              \AttributeTok{tip\_length =} \FloatTok{0.01}\NormalTok{)}


\NormalTok{p\_fast2}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-25-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{fig4G }\OtherTok{\textless{}{-}} \FunctionTok{plot\_grid}\NormalTok{(p\_fast, p\_fast2, }\AttributeTok{nrow =} \DecValTok{2}\NormalTok{, }\AttributeTok{ncol=}\DecValTok{1}\NormalTok{)}
\FunctionTok{ggsave}\NormalTok{(}\StringTok{"/Users/xl7/Documents/0.Projects/01.protein{-}seq{-}evo{-}v1/figs/panels/fig4\_violin5.pdf"}\NormalTok{, }
       \AttributeTok{plot =}\NormalTok{ fig4G, }\AttributeTok{width =} \DecValTok{3}\NormalTok{, }\AttributeTok{height =} \DecValTok{6}\NormalTok{, }\AttributeTok{dpi =} \DecValTok{300}\NormalTok{)}
\NormalTok{fig4G}
\end{Highlighting}
\end{Shaded}

\includegraphics{fig4_v2_files/figure-latex/unnamed-chunk-26-1.pdf}

\end{document}
